<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 4 Week 3- Working with Files Part 2 | 03-Week3.Rmd</title>
  <meta name="description" content=" 4 Week 3- Working with Files Part 2 | 03-Week3.Rmd" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content=" 4 Week 3- Working with Files Part 2 | 03-Week3.Rmd" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 4 Week 3- Working with Files Part 2 | 03-Week3.Rmd" />
  
  
  

<meta name="author" content="M. Armstrong" />


<meta name="date" content="2025-03-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="week-2--working-with-files.html"/>
<link rel="next" href="week-4--what-is-a-genetic-variant.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to Genomics in Natural Populations</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro to Genomics in Natural Populations</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#course-learning-objectives"><i class="fa fa-check"></i><b>1.1</b> Course Learning Objectives</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#meet-your-instructors"><i class="fa fa-check"></i><b>1.2</b> Meet your instructors!</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#getting-your-farm-account-set-up-class-specific"><i class="fa fa-check"></i><b>1.3</b> Getting your Farm Account set up (class specific)</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="week-1--welcome.html"><a href="week-1--welcome.html"><i class="fa fa-check"></i><b>2</b> Week 1- Welcome!</a>
<ul>
<li class="chapter" data-level="2.1" data-path="week-1--welcome.html"><a href="week-1--welcome.html#main-objectives"><i class="fa fa-check"></i><b>2.1</b> Main Objectives</a></li>
<li class="chapter" data-level="2.2" data-path="week-1--welcome.html"><a href="week-1--welcome.html#how-to-access-the-shell-via-farm-ondemand"><i class="fa fa-check"></i><b>2.2</b> How to access the shell via Farm OnDemand</a></li>
<li class="chapter" data-level="2.3" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-your-file-system"><i class="fa fa-check"></i><b>2.3</b> Navigating your file system</a></li>
<li class="chapter" data-level="2.4" data-path="week-1--welcome.html"><a href="week-1--welcome.html#shortcut-tab-completion"><i class="fa fa-check"></i><b>2.4</b> Shortcut: Tab Completion</a></li>
<li class="chapter" data-level="2.5" data-path="week-1--welcome.html"><a href="week-1--welcome.html#creating-our-own-directories"><i class="fa fa-check"></i><b>2.5</b> Creating our own directories</a></li>
<li class="chapter" data-level="2.6" data-path="week-1--welcome.html"><a href="week-1--welcome.html#full-vs-relative-paths"><i class="fa fa-check"></i><b>2.6</b> Full vs Relative Paths</a></li>
<li class="chapter" data-level="2.7" data-path="week-1--welcome.html"><a href="week-1--welcome.html#examining-the-contents-of-other-directories"><i class="fa fa-check"></i><b>2.7</b> Examining the contents of other directories</a></li>
<li class="chapter" data-level="2.8" data-path="week-1--welcome.html"><a href="week-1--welcome.html#group-work-activity--treasure-hunt"><i class="fa fa-check"></i><b>2.8</b> Group Work Activity- Treasure Hunt!</a></li>
<li class="chapter" data-level="2.9" data-path="week-1--welcome.html"><a href="week-1--welcome.html#key-points"><i class="fa fa-check"></i><b>2.9</b> Key Points</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html"><i class="fa fa-check"></i><b>3</b> Week 2- Working with Files</a>
<ul>
<li class="chapter" data-level="3.1" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#main-objectives-1"><i class="fa fa-check"></i><b>3.1</b> Main Objectives:</a></li>
<li class="chapter" data-level="3.2" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#navigational-shortcuts"><i class="fa fa-check"></i><b>3.2</b> Navigational shortcuts</a></li>
<li class="chapter" data-level="3.3" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#sequencing-pipeline"><i class="fa fa-check"></i><b>3.3</b> Sequencing Pipeline</a></li>
<li class="chapter" data-level="3.4" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#details-on-the-fastq-format"><i class="fa fa-check"></i><b>3.4</b> Details on the FASTQ format</a></li>
<li class="chapter" data-level="3.5" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#our-data-set-fastq-files"><i class="fa fa-check"></i><b>3.5</b> Our data set: FASTQ files</a></li>
<li class="chapter" data-level="3.6" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#wildcards"><i class="fa fa-check"></i><b>3.6</b> Wildcards</a></li>
<li class="chapter" data-level="3.7" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#command-history"><i class="fa fa-check"></i><b>3.7</b> Command History</a></li>
<li class="chapter" data-level="3.8" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#examining-files"><i class="fa fa-check"></i><b>3.8</b> Examining Files</a></li>
<li class="chapter" data-level="3.9" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#group-work-activity--examining-a-fastq-file"><i class="fa fa-check"></i><b>3.9</b> Group Work Activity- Examining a Fastq File</a></li>
<li class="chapter" data-level="3.10" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#key-points-1"><i class="fa fa-check"></i><b>3.10</b> Key Points</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html"><i class="fa fa-check"></i><b>4</b> Week 3- Working with Files Part 2</a>
<ul>
<li class="chapter" data-level="4.1" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#main-objectives-2"><i class="fa fa-check"></i><b>4.1</b> Main Objectives</a></li>
<li class="chapter" data-level="4.2" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#creating-moving-copying-and-removing"><i class="fa fa-check"></i><b>4.2</b> Creating, moving, copying, and removing</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#copying-files"><i class="fa fa-check"></i><b>4.2.1</b> Copying Files</a></li>
<li class="chapter" data-level="4.2.2" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#creating-directories"><i class="fa fa-check"></i><b>4.2.2</b> Creating Directories</a></li>
<li class="chapter" data-level="4.2.3" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#moving-renaming"><i class="fa fa-check"></i><b>4.2.3</b> Moving / Renaming</a></li>
<li class="chapter" data-level="4.2.4" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#file-permissions"><i class="fa fa-check"></i><b>4.2.4</b> File Permissions</a></li>
<li class="chapter" data-level="4.2.5" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#removing"><i class="fa fa-check"></i><b>4.2.5</b> Removing</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#searching-files"><i class="fa fa-check"></i><b>4.3</b> Searching files</a></li>
<li class="chapter" data-level="4.4" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#redirecting-output"><i class="fa fa-check"></i><b>4.4</b> Redirecting output</a></li>
<li class="chapter" data-level="4.5" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#file-extensions"><i class="fa fa-check"></i><b>4.5</b> File extensions</a></li>
<li class="chapter" data-level="4.6" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#writing-for-loops"><i class="fa fa-check"></i><b>4.6</b> Writing for loops</a></li>
<li class="chapter" data-level="4.7" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#using-basename-in-for-loops"><i class="fa fa-check"></i><b>4.7</b> Using Basename in for loops</a></li>
<li class="chapter" data-level="4.8" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#writing-scripts-and-working-with-data"><i class="fa fa-check"></i><b>4.8</b> Writing Scripts and Working with Data</a></li>
<li class="chapter" data-level="4.9" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#writing-scripts-bash"><i class="fa fa-check"></i><b>4.9</b> Writing scripts: Bash</a></li>
<li class="chapter" data-level="4.10" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#writing-scripts-slurm"><i class="fa fa-check"></i><b>4.10</b> Writing scripts: Slurm</a></li>
<li class="chapter" data-level="4.11" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#group-work-activity--writing-your-own-script"><i class="fa fa-check"></i><b>4.11</b> Group Work Activity- Writing your own script</a></li>
<li class="chapter" data-level="4.12" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#key-points-2"><i class="fa fa-check"></i><b>4.12</b> Key Points</a></li>
<li class="chapter" data-level="4.13" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#additional-information"><i class="fa fa-check"></i><b>4.13</b> Additional Information</a>
<ul>
<li class="chapter" data-level="4.13.1" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix"><i class="fa fa-check"></i><b>4.13.1</b> Uploading and Downloading Data to your Virtual Machine with scp - UNIX</a></li>
</ul></li>
<li class="chapter" data-level="4.14" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#more-ways-to-work-with-files"><i class="fa fa-check"></i><b>4.14</b> More ways to work with files</a>
<ul>
<li class="chapter" data-level="4.14.1" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#awk"><i class="fa fa-check"></i><b>4.14.1</b> Awk</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html"><i class="fa fa-check"></i><b>5</b> Week 4- What is a Genetic Variant?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#main-objectives-3"><i class="fa fa-check"></i><b>5.1</b> Main Objectives</a></li>
<li class="chapter" data-level="5.2" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#step-1-download-data"><i class="fa fa-check"></i><b>5.2</b> Step 1: Download data</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#getting-data-from-the-cloud"><i class="fa fa-check"></i><b>5.2.1</b> Getting data from the cloud</a></li>
<li class="chapter" data-level="5.2.2" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#tools-for-todays-class"><i class="fa fa-check"></i><b>5.2.2</b> tools for today’s class</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#step-1.2-raw-read-quality-control"><i class="fa fa-check"></i><b>5.3</b> Step 1.2: Raw read quality control</a></li>
<li class="chapter" data-level="5.4" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#step-2-trimming-to-remove-adapters"><i class="fa fa-check"></i><b>5.4</b> Step 2: Trimming to remove adapters</a></li>
<li class="chapter" data-level="5.5" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#step-3-building-an-index-of-our-genome"><i class="fa fa-check"></i><b>5.5</b> Step 3: Building an index of our genome</a></li>
<li class="chapter" data-level="5.6" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#step-4-map-reads-to-the-genome"><i class="fa fa-check"></i><b>5.6</b> Step 4: Map reads to the genome</a></li>
<li class="chapter" data-level="5.7" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#step-5-sam-to-bam-file-conversion"><i class="fa fa-check"></i><b>5.7</b> Step 5: sam to bam file conversion</a></li>
<li class="chapter" data-level="5.8" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#step-6-genotype-likelihoods"><i class="fa fa-check"></i><b>5.8</b> Step 6: Genotype likelihoods</a></li>
<li class="chapter" data-level="5.9" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#group-work-activity--calling-variants-at-a-larger-scale"><i class="fa fa-check"></i><b>5.9</b> Group Work Activity- Calling Variants at a Larger Scale</a></li>
<li class="chapter" data-level="5.10" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#key-points-3"><i class="fa fa-check"></i><b>5.10</b> Key Points</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html"><i class="fa fa-check"></i><b>6</b> Week 5- Welcome to the R environment!</a>
<ul>
<li class="chapter" data-level="6.1" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html#main-objectives-4"><i class="fa fa-check"></i><b>6.1</b> Main Objectives</a></li>
<li class="chapter" data-level="6.2" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html#orientation-to-r"><i class="fa fa-check"></i><b>6.2</b> Orientation to R</a></li>
<li class="chapter" data-level="6.3" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html#manipulating-a-vector-object"><i class="fa fa-check"></i><b>6.3</b> Manipulating a vector object</a></li>
<li class="chapter" data-level="6.4" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html#operations-act-on-each-element-of-a-vector"><i class="fa fa-check"></i><b>6.4</b> Operations act on each element of a vector:</a></li>
<li class="chapter" data-level="6.5" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html#operations-can-also-work-with-two-vectors"><i class="fa fa-check"></i><b>6.5</b> Operations can also work with two vectors:</a></li>
<li class="chapter" data-level="6.6" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html#a-few-tips-below-for-working-with-objects"><i class="fa fa-check"></i><b>6.6</b> A few tips below for working with objects:</a></li>
<li class="chapter" data-level="6.7" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html#characterizing-a-dataframe"><i class="fa fa-check"></i><b>6.7</b> Characterizing a dataframe</a></li>
<li class="chapter" data-level="6.8" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html#how-to-access-parts-of-the-data"><i class="fa fa-check"></i><b>6.8</b> How to access parts of the data:</a></li>
<li class="chapter" data-level="6.9" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html#data-manipulation"><i class="fa fa-check"></i><b>6.9</b> Data Manipulation</a></li>
<li class="chapter" data-level="6.10" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html#group-work-activity-practice-exploring-a-dataframe"><i class="fa fa-check"></i><b>6.10</b> Group Work Activity: practice exploring a dataframe</a></li>
<li class="chapter" data-level="6.11" data-path="week-5--welcome-to-the-r-environment.html"><a href="week-5--welcome-to-the-r-environment.html#key-points-4"><i class="fa fa-check"></i><b>6.11</b> Key Points</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="week-6--plotting-in-r.html"><a href="week-6--plotting-in-r.html"><i class="fa fa-check"></i><b>7</b> Week 6- Plotting in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="week-6--plotting-in-r.html"><a href="week-6--plotting-in-r.html#main-objectives-5"><i class="fa fa-check"></i><b>7.1</b> Main Objectives</a></li>
<li class="chapter" data-level="7.2" data-path="week-6--plotting-in-r.html"><a href="week-6--plotting-in-r.html#subsetting-datasets-logicals"><i class="fa fa-check"></i><b>7.2</b> Subsetting datasets &amp; logicals</a></li>
<li class="chapter" data-level="7.3" data-path="week-6--plotting-in-r.html"><a href="week-6--plotting-in-r.html#plotting-in-base-r"><i class="fa fa-check"></i><b>7.3</b> Plotting in Base R</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="week-6--plotting-in-r.html"><a href="week-6--plotting-in-r.html#scatterplots"><i class="fa fa-check"></i><b>7.3.1</b> Scatterplots</a></li>
<li class="chapter" data-level="7.3.2" data-path="week-6--plotting-in-r.html"><a href="week-6--plotting-in-r.html#histograms"><i class="fa fa-check"></i><b>7.3.2</b> Histograms</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="week-6--plotting-in-r.html"><a href="week-6--plotting-in-r.html#plotting-with-ggplot2"><i class="fa fa-check"></i><b>7.4</b> Plotting with ggplot2</a></li>
<li class="chapter" data-level="7.5" data-path="week-6--plotting-in-r.html"><a href="week-6--plotting-in-r.html#for-loops-and-the-apply-family-of-functions"><i class="fa fa-check"></i><b>7.5</b> For loops and the apply family of functions</a></li>
<li class="chapter" data-level="7.6" data-path="week-6--plotting-in-r.html"><a href="week-6--plotting-in-r.html#group-work-activity-palmer-penguins"><i class="fa fa-check"></i><b>7.6</b> Group Work Activity: Palmer Penguins!</a></li>
<li class="chapter" data-level="7.7" data-path="week-6--plotting-in-r.html"><a href="week-6--plotting-in-r.html#key-points-5"><i class="fa fa-check"></i><b>7.7</b> Key Points</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html"><i class="fa fa-check"></i><b>8</b> Week 7- Population Structure &amp; Sea Cucumbers</a>
<ul>
<li class="chapter" data-level="8.1" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#main-objectives-6"><i class="fa fa-check"></i><b>8.1</b> Main Objectives:</a></li>
<li class="chapter" data-level="8.2" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#download-the-data-in-terminal"><i class="fa fa-check"></i><b>8.2</b> Download the data in terminal</a></li>
<li class="chapter" data-level="8.3" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#moving-to-r"><i class="fa fa-check"></i><b>8.3</b> Moving to R:</a></li>
<li class="chapter" data-level="8.4" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#installing-packages"><i class="fa fa-check"></i><b>8.4</b> Installing Packages</a></li>
<li class="chapter" data-level="8.5" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#set-working-directory"><i class="fa fa-check"></i><b>8.5</b> Set Working Directory</a></li>
<li class="chapter" data-level="8.6" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#read-vcf-file"><i class="fa fa-check"></i><b>8.6</b> read vcf file</a></li>
<li class="chapter" data-level="8.7" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#read-in-the-metadata"><i class="fa fa-check"></i><b>8.7</b> read in the metadata</a></li>
<li class="chapter" data-level="8.8" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#population-structure-with-tess3r"><i class="fa fa-check"></i><b>8.8</b> population structure with tess3r</a></li>
<li class="chapter" data-level="8.9" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#how-do-we-know-which-k-to-pick"><i class="fa fa-check"></i><b>8.9</b> How do we know which K to pick?</a></li>
<li class="chapter" data-level="8.10" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#visualizing-population-structure-with-pophelper"><i class="fa fa-check"></i><b>8.10</b> Visualizing population structure with pophelper!</a></li>
<li class="chapter" data-level="8.11" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#adding-more-meaning-to-our-plot"><i class="fa fa-check"></i><b>8.11</b> Adding more meaning to our plot</a></li>
<li class="chapter" data-level="8.12" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#group-work-activity--adding-more-context-to-our-plot"><i class="fa fa-check"></i><b>8.12</b> Group Work Activity- Adding more context to our plot</a></li>
<li class="chapter" data-level="8.13" data-path="week-7--population-structure-sea-cucumbers.html"><a href="week-7--population-structure-sea-cucumbers.html#key-points-6"><i class="fa fa-check"></i><b>8.13</b> Key Points</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="week-8-pcas-mosquitos.html"><a href="week-8-pcas-mosquitos.html"><i class="fa fa-check"></i><b>9</b> Week 8: PCAs &amp; Mosquitos</a>
<ul>
<li class="chapter" data-level="9.1" data-path="week-8-pcas-mosquitos.html"><a href="week-8-pcas-mosquitos.html#main-objectives-7"><i class="fa fa-check"></i><b>9.1</b> Main Objectives:</a></li>
<li class="chapter" data-level="9.2" data-path="week-8-pcas-mosquitos.html"><a href="week-8-pcas-mosquitos.html#download-the-data-in-terminal-1"><i class="fa fa-check"></i><b>9.2</b> Download the data in terminal</a></li>
<li class="chapter" data-level="9.3" data-path="week-8-pcas-mosquitos.html"><a href="week-8-pcas-mosquitos.html#moving-to-r-installing-packages"><i class="fa fa-check"></i><b>9.3</b> Moving to R &amp; Installing Packages</a></li>
<li class="chapter" data-level="9.4" data-path="week-8-pcas-mosquitos.html"><a href="week-8-pcas-mosquitos.html#set-working-directory-1"><i class="fa fa-check"></i><b>9.4</b> Set Working Directory</a></li>
<li class="chapter" data-level="9.5" data-path="week-8-pcas-mosquitos.html"><a href="week-8-pcas-mosquitos.html#read-vcf-file-1"><i class="fa fa-check"></i><b>9.5</b> read vcf file</a></li>
<li class="chapter" data-level="9.6" data-path="week-8-pcas-mosquitos.html"><a href="week-8-pcas-mosquitos.html#read-in-metadata"><i class="fa fa-check"></i><b>9.6</b> Read in Metadata</a></li>
<li class="chapter" data-level="9.7" data-path="week-8-pcas-mosquitos.html"><a href="week-8-pcas-mosquitos.html#prepare-for-plotting"><i class="fa fa-check"></i><b>9.7</b> Prepare for Plotting</a></li>
<li class="chapter" data-level="9.8" data-path="week-8-pcas-mosquitos.html"><a href="week-8-pcas-mosquitos.html#group-work-activity--what-can-the-other-pc-axes-tell-us"><i class="fa fa-check"></i><b>9.8</b> Group Work Activity- What can the other PC axes tell us?</a></li>
<li class="chapter" data-level="9.9" data-path="week-8-pcas-mosquitos.html"><a href="week-8-pcas-mosquitos.html#key-point"><i class="fa fa-check"></i><b>9.9</b> Key Point</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="week-9-fst-outlier-analysis-and-chunky-quails.html"><a href="week-9-fst-outlier-analysis-and-chunky-quails.html"><i class="fa fa-check"></i><b>10</b> Week 9: F<sub>st</sub> outlier analysis and chunky quails</a>
<ul>
<li class="chapter" data-level="10.1" data-path="week-9-fst-outlier-analysis-and-chunky-quails.html"><a href="week-9-fst-outlier-analysis-and-chunky-quails.html#main-objectives-8"><i class="fa fa-check"></i><b>10.1</b> Main Objectives:</a></li>
<li class="chapter" data-level="10.2" data-path="week-9-fst-outlier-analysis-and-chunky-quails.html"><a href="week-9-fst-outlier-analysis-and-chunky-quails.html#download-the-data"><i class="fa fa-check"></i><b>10.2</b> Download the data</a></li>
<li class="chapter" data-level="10.3" data-path="week-9-fst-outlier-analysis-and-chunky-quails.html"><a href="week-9-fst-outlier-analysis-and-chunky-quails.html#downloading-r-packages"><i class="fa fa-check"></i><b>10.3</b> Downloading R Packages</a></li>
<li class="chapter" data-level="10.4" data-path="week-9-fst-outlier-analysis-and-chunky-quails.html"><a href="week-9-fst-outlier-analysis-and-chunky-quails.html#finding-outliers-using-outflank"><i class="fa fa-check"></i><b>10.4</b> Finding outliers using OutFLANK</a></li>
<li class="chapter" data-level="10.5" data-path="week-9-fst-outlier-analysis-and-chunky-quails.html"><a href="week-9-fst-outlier-analysis-and-chunky-quails.html#pulling-snp-data-from-the-gds-file"><i class="fa fa-check"></i><b>10.5</b> Pulling SNP data from the gds file</a></li>
<li class="chapter" data-level="10.6" data-path="week-9-fst-outlier-analysis-and-chunky-quails.html"><a href="week-9-fst-outlier-analysis-and-chunky-quails.html#group-work-activity-pca-outflank"><i class="fa fa-check"></i><b>10.6</b> Group Work Activity: PCA &amp; Outflank</a></li>
<li class="chapter" data-level="10.7" data-path="week-9-fst-outlier-analysis-and-chunky-quails.html"><a href="week-9-fst-outlier-analysis-and-chunky-quails.html#key-points-7"><i class="fa fa-check"></i><b>10.7</b> Key Points</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">03-Week3.Rmd</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="week-3--working-with-files-part-2" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number"> 4</span> Week 3- Working with Files Part 2<a href="week-3--working-with-files-part-2.html#week-3--working-with-files-part-2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>We now know how to switch directories, run programs, look at the
contents of directories and files, but how do we manipulate files?</p>
<div id="main-objectives-2" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Main Objectives<a href="week-3--working-with-files-part-2.html#main-objectives-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Learn how to create, copy and delete files/directories</li>
<li>Learn how to modify permissions for a file</li>
<li>Construct command pipelines with two or more stages</li>
<li>Use <code>for</code> loops to run the same command for several input files.</li>
<li>Practice writing scripts to work with data in terminal</li>
</ul>
</div>
<div id="creating-moving-copying-and-removing" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Creating, moving, copying, and removing<a href="week-3--working-with-files-part-2.html#creating-moving-copying-and-removing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now we can move around in the file structure, look at files, and search files. But what if we want to copy files or move them around or get rid of them? Most of the time, you can do these sorts of file manipulations without the command line, but there will be some cases (like when you’re working with a remote computer like we are for this lesson) where it will be impossible. You’ll also find that you may be working with hundreds of files and want to do similar manipulations to all of those files. In cases like this, it’s much faster to do these operations at the command line.</p>
<div id="copying-files" class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Copying Files<a href="week-3--working-with-files-part-2.html#copying-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>When working with computational data, it’s important to keep a safe copy of that data that can’t be accidentally overwritten or deleted.
For this lesson, our raw data is our FASTQ files. We don’t want to accidentally change the original files, so we’ll make a copy of them
and change the file permissions so that we can read from, but not write to, the files.</p>
<p>First, let’s make a copy of one of our FASTQ files using the <code>cp</code> command.</p>
<p>Navigate to the your directory, then move into your Week2/untrimmed_fastq directory with the fastq files we worked with last week. Once you are there, type:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb1-1"><a href="week-3--working-with-files-part-2.html#cb1-1" tabindex="-1"></a>$ cp SRR098026.fastq SRR098026-copy.fastq</span>
<span id="cb1-2"><a href="week-3--working-with-files-part-2.html#cb1-2" tabindex="-1"></a>$ ls -F</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb2-1"><a href="week-3--working-with-files-part-2.html#cb2-1" tabindex="-1"></a>SRR097977.fastq  SRR098026-copy.fastq  SRR098026.fastq</span></code></pre></div>
<p>We now have two copies of the <code>SRR098026.fastq</code> file, one of them named <code>SRR098026-copy.fastq</code>. We’ll move this file to a new directory
called <code>backup</code> where we’ll store our backup data files.</p>
</div>
<div id="creating-directories" class="section level3 hasAnchor" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Creating Directories<a href="week-3--working-with-files-part-2.html#creating-directories" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>mkdir</code> command is used to make a directory. Enter <code>mkdir</code>
followed by a space, then the directory name you want to create.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb3-1"><a href="week-3--working-with-files-part-2.html#cb3-1" tabindex="-1"></a>$ mkdir backup</span></code></pre></div>
</div>
<div id="moving-renaming" class="section level3 hasAnchor" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> Moving / Renaming<a href="week-3--working-with-files-part-2.html#moving-renaming" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can now move our backup file to this directory. We can
move files around using the command <code>mv</code>:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb4-1"><a href="week-3--working-with-files-part-2.html#cb4-1" tabindex="-1"></a>$ mv SRR098026-copy.fastq backup</span>
<span id="cb4-2"><a href="week-3--working-with-files-part-2.html#cb4-2" tabindex="-1"></a>$ ls backup</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb5-1"><a href="week-3--working-with-files-part-2.html#cb5-1" tabindex="-1"></a>SRR098026-copy.fastq</span></code></pre></div>
<p>The <code>mv</code> command is also how you rename files. Let’s rename this file to make it clear that this is a backup:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb6-1"><a href="week-3--working-with-files-part-2.html#cb6-1" tabindex="-1"></a>$ cd backup</span>
<span id="cb6-2"><a href="week-3--working-with-files-part-2.html#cb6-2" tabindex="-1"></a>$ mv SRR098026-copy.fastq SRR098026-backup.fastq</span>
<span id="cb6-3"><a href="week-3--working-with-files-part-2.html#cb6-3" tabindex="-1"></a>$ ls</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb7-1"><a href="week-3--working-with-files-part-2.html#cb7-1" tabindex="-1"></a>SRR098026-backup.fastq</span></code></pre></div>
</div>
<div id="file-permissions" class="section level3 hasAnchor" number="4.2.4">
<h3><span class="header-section-number">4.2.4</span> File Permissions<a href="week-3--working-with-files-part-2.html#file-permissions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We’ve now made a backup copy of our file, but just because we have two copies, it doesn’t make us safe. We can still accidentally delete or
overwrite both copies. To make sure we can’t accidentally mess up this backup file, we’re going to change the permissions on the file so
that we’re only allowed to read (i.e. view) the file, not write to it (i.e. make new changes).</p>
<p>View the current permissions on a file using the <code>-l</code> (long) flag for the <code>ls</code> command:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb8-1"><a href="week-3--working-with-files-part-2.html#cb8-1" tabindex="-1"></a>$ ls -l</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb9-1"><a href="week-3--working-with-files-part-2.html#cb9-1" tabindex="-1"></a>-rw-rw-r-- 1 margeno margeno 43K Apr 6 12:15 SRR098026-backup.fastq</span></code></pre></div>
<p>The first part of the output for the <code>-l</code> flag gives you information about the file’s current permissions. There are ten slots in the
permissions list. The first character in this list is related to file type, not permissions, so we’ll ignore it for now. The next three
characters relate to the permissions that the file owner has, the next three relate to the permissions for group members, and the final
three characters specify what other users outside of your group can do with the file. We’re going to concentrate on the three positions
that deal with your permissions (as the file owner).</p>
<div class="float">
<img src="figs/rwx_figure.png" alt="Permissions" />
<div class="figcaption">Permissions</div>
</div>
<p>Here the three positions that relate to the file owner are <code>rw-</code>. The <code>r</code> means that you have permission to read the file, the <code>w</code>
indicates that you have permission to write to (i.e. make changes to) the file, and the third position is a <code>-</code>, indicating that you
don’t have permission to carry out the ability encoded by that space (this is the space where <code>x</code> or executable ability is stored.</p>
<p>Our goal for now is to change permissions on this file so that you no longer have <code>w</code> or write permissions. We can do this using the <code>chmod</code> (change mode) command and subtracting (<code>-</code>) the write permission <code>-w</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb10-1"><a href="week-3--working-with-files-part-2.html#cb10-1" tabindex="-1"></a>$ chmod -w SRR098026-backup.fastq</span>
<span id="cb10-2"><a href="week-3--working-with-files-part-2.html#cb10-2" tabindex="-1"></a>$ ls -l </span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb11-1"><a href="week-3--working-with-files-part-2.html#cb11-1" tabindex="-1"></a>-r--r--r-- 1 dcuser dcuser 43332 Nov 15 23:02 SRR098026-backup.fastq</span></code></pre></div>
<p>Chmod can also change the permission to only the user (u), group (g), and/or other (o). Let’s add reading (r), writing (x), and execute (x) permissions to user group.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb12-1"><a href="week-3--working-with-files-part-2.html#cb12-1" tabindex="-1"></a>$ chmod u=rwx SRR098026-backup.fastq</span>
<span id="cb12-2"><a href="week-3--working-with-files-part-2.html#cb12-2" tabindex="-1"></a>$ ls -l </span></code></pre></div>
</div>
<div id="removing" class="section level3 hasAnchor" number="4.2.5">
<h3><span class="header-section-number">4.2.5</span> Removing<a href="week-3--working-with-files-part-2.html#removing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To prove to ourselves that you no longer have the ability to modify this file, try deleting it with the <code>rm</code> command:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb13-1"><a href="week-3--working-with-files-part-2.html#cb13-1" tabindex="-1"></a>$ rm SRR098026-backup.fastq</span></code></pre></div>
<p>You’ll be asked if you want to override your file permissions:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb14-1"><a href="week-3--working-with-files-part-2.html#cb14-1" tabindex="-1"></a>rm: remove write-protected regular file ‘SRR098026-backup.fastq’? </span></code></pre></div>
<p>You should enter <code>n</code> for no. If you enter <code>n</code> (for no), the file will not be deleted. If you enter <code>y</code>, you will delete the file. This gives us an extra
measure of security, as there is one more step between us and deleting our data files.</p>
<p><strong>Important</strong>: The <code>rm</code> command permanently removes the file. Be careful with this command. It doesn’t just nicely put the files in the Trash. They’re really gone.</p>
<p>By default, <code>rm</code> will not delete directories. You can tell <code>rm</code> to
delete a directory using the <code>-r</code> (recursive) option. Let’s delete the backup directory
we just made.</p>
<p>Enter the following command:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb15-1"><a href="week-3--working-with-files-part-2.html#cb15-1" tabindex="-1"></a>$ cd ..</span>
<span id="cb15-2"><a href="week-3--working-with-files-part-2.html#cb15-2" tabindex="-1"></a>$ rm -r backup</span></code></pre></div>
<p>This will delete not only the directory, but all files within the directory. If you have write-protected files in the directory,
you will be asked whether you want to override your permission settings.</p>
<blockquote>
<h3 id="class-exercise-1" class="hasAnchor">Class Exercise 1<a href="week-3--working-with-files-part-2.html#class-exercise-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Starting in ‘yourdirectory’, navigate to ‘cd /week2/untrimmed_fastq’ and do the following:</p>
<ol style="list-style-type: decimal">
<li><p>Make sure that you have deleted your backup directory and all files it contains since we are going to do it again but with less commands!</p></li>
<li><p>Create a backup of each of your FASTQ files using <code>cp</code>. (Note: You’ll need to do this individually for each of the two FASTQ files. We haven’t earned yet how to do this with a wildcard.)</p></li>
<li><p>Use a wildcard to move all of your backup files to a new backup directory.</p></li>
<li><p>Change the permissions on all of your backup files to be write-protected.</p></li>
</ol>
</blockquote>
</div>
</div>
<div id="searching-files" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Searching files<a href="week-3--working-with-files-part-2.html#searching-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We discussed in a previous episode how to search within a file using <code>less</code>. We can also
search within files without even opening them, using <code>grep</code>. <code>grep</code> is a command-line
utility for searching plain-text files for lines matching a specific set of
characters (sometimes called a string) or a particular pattern
(which can be specified using something called regular expressions). We’re not going to work with
regular expressions in this lesson, and are instead going to specify the strings
we are searching for.
Let’s give it a try!</p>
<blockquote>
<h3 id="nucleotide-abbreviations" class="hasAnchor">Nucleotide abbreviations<a href="week-3--working-with-files-part-2.html#nucleotide-abbreviations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The four nucleotides that appear in DNA are abbreviated <code>A</code>, <code>C</code>, <code>T</code> and <code>G</code>.
Unknown nucleotides are represented with the letter <code>N</code>. An <code>N</code> appearing
in a sequencing file represents a position where the sequencing machine was not able to
confidently determine the nucleotide in that position. You can think of an <code>N</code> as being aNy
nucleotide at that position in the DNA sequence.</p>
</blockquote>
<p>We’ll search for strings inside of our fastq files. Let’s first make sure we are in the correct
directory:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb16-1"><a href="week-3--working-with-files-part-2.html#cb16-1" tabindex="-1"></a>$ cd /group/rbaygrp/eve198-genomics/yourdirectory/week2/untrimmed_fastq</span></code></pre></div>
<p>Suppose we want to see how many reads in our file have really bad segments containing 10 consecutive unknown nucleotides (Ns).To determine the quality of our data, we’re going to be manually searching for strings of <code>N</code>s within our sequence results to illustrate some principles of file searching. It can be really useful to do this type of searching to get a feel for the quality of your sequencing results, however, in your research you will most likely use a bioinformatics tool that has a built-in program for filtering out low-quality reads. You’ll learn how to use one such tool in <a href="https://datacarpentry.org/wrangling-genomics/02-quality-control/index.html">a later lesson</a>.</p>
<p>Let’s search for the string NNNNNNNNNN in the SRR098026 file:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb17-1"><a href="week-3--working-with-files-part-2.html#cb17-1" tabindex="-1"></a>$ grep NNNNNNNNNN SRR098026.fastq</span></code></pre></div>
<p>This command returns a lot of output to the terminal. Every single line in the SRR098026
file that contains at least 10 consecutive Ns is printed to the terminal, regardless of how long or short the file is.
We may be interested not only in the actual sequence which contains this string, but
in the name (or identifier) of that sequence. We discussed in a previous lesson
that the identifier line immediately precedes the nucleotide sequence for each read
in a FASTQ file. We may also want to inspect the quality scores associated with
each of these reads. To get all of this information, we will return the line
immediately before each match and the two lines immediately after each match.</p>
<p>We can use the <code>-B</code> argument for grep to return a specific number of lines before
each match. The <code>-A</code> argument returns a specific number of lines after each matching line. Here we want the line <em>before</em> and the two lines <em>after</em> each
matching line, so we add <code>-B1 -A2</code> to our grep command:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb18-1"><a href="week-3--working-with-files-part-2.html#cb18-1" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq</span></code></pre></div>
<p>One of the sets of lines returned by this command is:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb19-1"><a href="week-3--working-with-files-part-2.html#cb19-1" tabindex="-1"></a>@SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb19-2"><a href="week-3--working-with-files-part-2.html#cb19-2" tabindex="-1"></a>CNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span>
<span id="cb19-3"><a href="week-3--working-with-files-part-2.html#cb19-3" tabindex="-1"></a>+SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb19-4"><a href="week-3--working-with-files-part-2.html#cb19-4" tabindex="-1"></a>#!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<blockquote>
<h3 id="class-exercise-2" class="hasAnchor">Class Exercise 2<a href="week-3--working-with-files-part-2.html#class-exercise-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Search for the sequence <code>GNATNACCACTTCC</code> in the <code>SRR098026.fastq</code> file.
Have your search return all matching lines and the name (or identifier) for each sequence
that contains a match.</p></li>
<li><p>Search for the sequence <code>AAGTT</code> in both FASTQ files.
Have your search return all matching lines and the name (or identifier) for each sequence
that contains a match.</p></li>
</ol>
</blockquote>
</div>
<div id="redirecting-output" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Redirecting output<a href="week-3--working-with-files-part-2.html#redirecting-output" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>grep</code> allowed us to identify sequences in our FASTQ files that match a particular pattern.
All of these sequences were printed to our terminal screen, but in order to work with these
sequences and perform other operations on them, we will need to capture that output in some
way.</p>
<p>We can do this with something called “redirection”. The idea is that
we are taking what would ordinarily be printed to the terminal screen and redirecting it to another location.
In our case, we want to print this information to a file so that we can look at it later and
use other commands to analyze this data.</p>
<p>The command for redirecting output to a file is <code>&gt;</code>.</p>
<p>Let’s try out this command and copy all the records (including all four lines of each record)
in our FASTQ files that contain
‘NNNNNNNNNN’ to another file called <code>bad_reads.txt</code>.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb20-1"><a href="week-3--working-with-files-part-2.html#cb20-1" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq &gt; bad_reads.txt</span></code></pre></div>
</div>
<div id="file-extensions" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> File extensions<a href="week-3--working-with-files-part-2.html#file-extensions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You might be confused about why we’re naming our output file with a <code>.txt</code> extension. After all, it will be holding FASTQ formatted data that we’re extracting from our FASTQ files. Won’t it also be a FASTQ file? The answer is, yes - it will be a FASTQ file and it would make sense to name it with a <code>.fastq</code> extension. However, using a <code>.fastq</code> extension will lead us to problems when we move to using wildcards later in this episode. We’ll point out where this becomes important. For now, it’s good that you’re thinking about file extensions!</p>
<p>The prompt should sit there a little bit, and then it should look like nothing
happened. But type <code>ls</code>. You should see a new file called <code>bad_reads.txt</code>.</p>
<p>We can check the number of lines in our new file using a command called <code>wc</code>.
<code>wc</code> stands for <strong>word count</strong>. This command counts the number of words, lines, and characters
in a file. The FASTQ file may change over time, so given the potential for updates we are using a file downloaded in 2020.</p>
<p>The command wc gives the following output:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb21-1"><a href="week-3--working-with-files-part-2.html#cb21-1" tabindex="-1"></a>$ wc bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb22-1"><a href="week-3--working-with-files-part-2.html#cb22-1" tabindex="-1"></a>  537  1073 23217 bad_reads.txt</span></code></pre></div>
<p>This will tell us the number of lines, words and characters in the file. If we
want only the number of lines, we can use the <code>-l</code> flag for <code>lines</code>.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb23-1"><a href="week-3--working-with-files-part-2.html#cb23-1" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb24-1"><a href="week-3--working-with-files-part-2.html#cb24-1" tabindex="-1"></a>537 bad_reads.txt</span></code></pre></div>
<p>We might want to search multiple FASTQ files for sequences that match our search pattern.
However, we need to be careful, because each time we use the <code>&gt;</code> command to redirect output
to a file, the new output will replace the output that was already present in the file.
This is called “overwriting” and, just like you don’t want to overwrite your video recording
of your kid’s first birthday party, you also want to avoid overwriting your data files.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb25-1"><a href="week-3--working-with-files-part-2.html#cb25-1" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq &gt; bad_reads.txt</span>
<span id="cb25-2"><a href="week-3--working-with-files-part-2.html#cb25-2" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb26-1"><a href="week-3--working-with-files-part-2.html#cb26-1" tabindex="-1"></a>537 bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb27-1"><a href="week-3--working-with-files-part-2.html#cb27-1" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR097977.fastq &gt; bad_reads.txt</span>
<span id="cb27-2"><a href="week-3--working-with-files-part-2.html#cb27-2" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb28-1"><a href="week-3--working-with-files-part-2.html#cb28-1" tabindex="-1"></a>0 bad_reads.txt</span></code></pre></div>
<p>Here, the output of our second call to <code>wc</code> shows that we no longer have any lines in our <code>bad_reads.txt</code> file. This is
because the second file we searched (<code>SRR097977.fastq</code>) does not contain any lines that match our
search sequence. So our file was overwritten and is now empty.</p>
<p>We can avoid overwriting our files by using the command <code>&gt;&gt;</code>. <code>&gt;&gt;</code> is known as the “append redirect” and will
append new output to the end of a file, rather than overwriting it.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb29-1"><a href="week-3--working-with-files-part-2.html#cb29-1" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq &gt; bad_reads.txt</span>
<span id="cb29-2"><a href="week-3--working-with-files-part-2.html#cb29-2" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb30-1"><a href="week-3--working-with-files-part-2.html#cb30-1" tabindex="-1"></a>537 bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb31-1"><a href="week-3--working-with-files-part-2.html#cb31-1" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR097977.fastq &gt;&gt; bad_reads.txt</span>
<span id="cb31-2"><a href="week-3--working-with-files-part-2.html#cb31-2" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb32-1"><a href="week-3--working-with-files-part-2.html#cb32-1" tabindex="-1"></a>537 bad_reads.txt</span></code></pre></div>
<p>The output of our second call to <code>wc</code> shows that we have not overwritten our original data.</p>
<p>We can also do this with a single line of code by using a wildcard:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb33-1"><a href="week-3--working-with-files-part-2.html#cb33-1" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN *.fastq &gt; bad_reads.txt</span>
<span id="cb33-2"><a href="week-3--working-with-files-part-2.html#cb33-2" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb34-1"><a href="week-3--working-with-files-part-2.html#cb34-1" tabindex="-1"></a>537 bad_reads.txt</span></code></pre></div>
<blockquote>
<h3 id="file-extensions---part-2" class="hasAnchor">File extensions - part 2<a href="week-3--working-with-files-part-2.html#file-extensions---part-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This is where we would have trouble if we were naming our output file with a <code>.fastq</code> extension.
If we already had a file called <code>bad_reads.fastq</code> (from our previous <code>grep</code> practice)
and then ran the command above using a <code>.fastq</code> extension instead of a <code>.txt</code> extension, <code>grep</code>
would give us a warning.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb35-1"><a href="week-3--working-with-files-part-2.html#cb35-1" tabindex="-1"></a>grep -B1 -A2 NNNNNNNNNN *.fastq &gt; bad_reads.fastq</span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb36-1"><a href="week-3--working-with-files-part-2.html#cb36-1" tabindex="-1"></a>grep: input file ‘bad_reads.fastq’ is also the output</span></code></pre></div>
<p><code>grep</code> is letting you know that the output file <code>bad_reads.fastq</code> is also included in your
<code>grep</code> call because it matches the <code>*.fastq</code> pattern. Be careful with this as it can lead to
some unintended results.</p>
</blockquote>
<p>Since we might have multiple different criteria we want to search for,
creating a new output file each time has the potential to clutter up our workspace. We also
thus far haven’t been interested in the actual contents of those files, only in the number of
reads that we’ve found. We created the files to store the reads and then counted the lines in
the file to see how many reads matched our criteria. There’s a way to do this, however, that
doesn’t require us to create these intermediate files - the pipe command (<code>|</code>).</p>
<p>What <code>|</code> does is take the output that is scrolling by on the terminal and uses that output as input to another command.
When our output was scrolling by, we might have wished we could slow it down and
look at it, like we can with <code>less</code>. Well it turns out that we can! We can redirect our output
from our <code>grep</code> call through the <code>less</code> command.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb37-1"><a href="week-3--working-with-files-part-2.html#cb37-1" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq | less</span></code></pre></div>
<p>We can now see the output from our <code>grep</code> call within the <code>less</code> interface. We can use the up and down arrows
to scroll through the output and use <code>q</code> to exit <code>less</code>.</p>
<p>If we don’t want to create a file before counting lines of output from our <code>grep</code> search, we could directly pipe
the output of the grep search to the command <code>wc -l</code>. This can be helpful for investigating your output if you are not sure
you would like to save it to a file.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb38-1"><a href="week-3--working-with-files-part-2.html#cb38-1" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq | wc -l </span></code></pre></div>
<p>Because we asked <code>grep</code> for all four lines of each FASTQ record, we need to divide the output by
four to get the number of sequences that match our search pattern. Since 802 / 4 = 200.5 and we
are expecting an integer number of records, there is something added or missing in <code>bad_reads.txt</code>.
If we explore <code>bad_reads.txt</code> using <code>less</code>, we might be able to notice what is causing the uneven
number of lines. Luckily, this issue happens by the end of the file so we can also spot it with <code>tail</code>.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb39-1"><a href="week-3--working-with-files-part-2.html#cb39-1" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq &gt; bad_reads.txt</span>
<span id="cb39-2"><a href="week-3--working-with-files-part-2.html#cb39-2" tabindex="-1"></a>$ tail bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb40-1"><a href="week-3--working-with-files-part-2.html#cb40-1" tabindex="-1"></a>@SRR098026.133 HWUSI-EAS1599_1:2:1:0:1978 length=35</span>
<span id="cb40-2"><a href="week-3--working-with-files-part-2.html#cb40-2" tabindex="-1"></a>ANNNNNNNNNTTCAGCGACTNNNNNNNNNNGTNGN</span>
<span id="cb40-3"><a href="week-3--working-with-files-part-2.html#cb40-3" tabindex="-1"></a>+SRR098026.133 HWUSI-EAS1599_1:2:1:0:1978 length=35</span>
<span id="cb40-4"><a href="week-3--working-with-files-part-2.html#cb40-4" tabindex="-1"></a>#!!!!!!!!!##########!!!!!!!!!!##!#!</span>
<span id="cb40-5"><a href="week-3--working-with-files-part-2.html#cb40-5" tabindex="-1"></a>--</span>
<span id="cb40-6"><a href="week-3--working-with-files-part-2.html#cb40-6" tabindex="-1"></a>--</span>
<span id="cb40-7"><a href="week-3--working-with-files-part-2.html#cb40-7" tabindex="-1"></a>@SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb40-8"><a href="week-3--working-with-files-part-2.html#cb40-8" tabindex="-1"></a>CNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span>
<span id="cb40-9"><a href="week-3--working-with-files-part-2.html#cb40-9" tabindex="-1"></a>+SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb40-10"><a href="week-3--working-with-files-part-2.html#cb40-10" tabindex="-1"></a>#!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<p>The fifth and six lines in the output display “–” which is the default action for <code>grep</code> to separate groups of
lines matching the pattern, and indicate groups of lines which did not match the pattern so are not displayed.
To fix this issue, we can redirect the output of grep to a second instance of <code>grep</code> as follows.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb41-1"><a href="week-3--working-with-files-part-2.html#cb41-1" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq | grep -v &#39;^--&#39; &gt; bad_reads.fastq</span>
<span id="cb41-2"><a href="week-3--working-with-files-part-2.html#cb41-2" tabindex="-1"></a>tail bad_reads.fastq</span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb42-1"><a href="week-3--working-with-files-part-2.html#cb42-1" tabindex="-1"></a>+SRR098026.132 HWUSI-EAS1599_1:2:1:0:320 length=35</span>
<span id="cb42-2"><a href="week-3--working-with-files-part-2.html#cb42-2" tabindex="-1"></a>#!!!!!!!!!##########!!!!!!!!!!##!#!</span>
<span id="cb42-3"><a href="week-3--working-with-files-part-2.html#cb42-3" tabindex="-1"></a>@SRR098026.133 HWUSI-EAS1599_1:2:1:0:1978 length=35</span>
<span id="cb42-4"><a href="week-3--working-with-files-part-2.html#cb42-4" tabindex="-1"></a>ANNNNNNNNNTTCAGCGACTNNNNNNNNNNGTNGN</span>
<span id="cb42-5"><a href="week-3--working-with-files-part-2.html#cb42-5" tabindex="-1"></a>+SRR098026.133 HWUSI-EAS1599_1:2:1:0:1978 length=35</span>
<span id="cb42-6"><a href="week-3--working-with-files-part-2.html#cb42-6" tabindex="-1"></a>#!!!!!!!!!##########!!!!!!!!!!##!#!</span>
<span id="cb42-7"><a href="week-3--working-with-files-part-2.html#cb42-7" tabindex="-1"></a>@SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb42-8"><a href="week-3--working-with-files-part-2.html#cb42-8" tabindex="-1"></a>CNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span>
<span id="cb42-9"><a href="week-3--working-with-files-part-2.html#cb42-9" tabindex="-1"></a>+SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb42-10"><a href="week-3--working-with-files-part-2.html#cb42-10" tabindex="-1"></a>#!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<p>The <code>-v</code> option in the second <code>grep</code> search stands for <code>--invert-match</code> meaning <code>grep</code> will now only display the
lines which do not match the searched pattern, in this case <code>'^--'</code>. The caret (<code>^</code>) is an <strong>anchoring</strong>
character matching the beginning of the line, and the pattern has to be enclose by single quotes so <code>grep</code> does
not interpret the pattern as an extended option (starting with –).</p>
<blockquote>
<h3 id="custom-grep-control" class="hasAnchor">Custom <code>grep</code> control<a href="week-3--working-with-files-part-2.html#custom-grep-control" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Use <code>man grep</code> to read more about other options to customize the output of <code>grep</code> including extended options,
anchoring characters, and much more.</p>
</blockquote>
<p>Redirecting output is often not intuitive, and can take some time to get used to. Once you’re
comfortable with redirection, however, you’ll be able to combine any number of commands to
do all sorts of exciting things with your data!</p>
<p>None of the command line programs we’ve been learning
do anything all that impressive on their own, but when you start chaining
them together, you can do some really powerful things very
efficiently.</p>
</div>
<div id="writing-for-loops" class="section level2 hasAnchor" number="4.6">
<h2><span class="header-section-number">4.6</span> Writing for loops<a href="week-3--working-with-files-part-2.html#writing-for-loops" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Loops are key to productivity improvements through automation as they allow us to execute commands repeatedly.
Similar to wildcards and tab completion, using loops also reduces the amount of typing (and typing mistakes).
Loops are helpful when performing operations on groups of sequencing files, such as unzipping or trimming multiple
files. We will use loops for these purposes in subsequent analyses, but will cover the basics of them for now.</p>
<p>When the shell sees the keyword <code>for</code>, it knows to repeat a command (or group of commands) once for each item in a list.
Each time the loop runs (called an iteration), an item in the list is assigned in sequence to the <strong>variable</strong>, and
the commands inside the loop are executed, before moving on to the next item in the list. Inside the loop, we call for
the variable’s value by putting <code>$</code> in front of it. The <code>$</code> tells the shell interpreter to treat the <strong>variable</strong>
as a variable name and substitute its value in its place, rather than treat it as text or an external command. In shell programming, this is usually called “expanding” the variable.</p>
<p>Sometimes, we want to expand a variable without any whitespace to its right.
Suppose we have a variable named <code>foo</code> that contains the text <code>abc</code>, and would
like to expand <code>foo</code> to create the text <code>abcEFG</code>.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb43-1"><a href="week-3--working-with-files-part-2.html#cb43-1" tabindex="-1"></a>$ foo=abc</span>
<span id="cb43-2"><a href="week-3--working-with-files-part-2.html#cb43-2" tabindex="-1"></a>$ echo foo is $foo</span>
<span id="cb43-3"><a href="week-3--working-with-files-part-2.html#cb43-3" tabindex="-1"></a>foo is abc</span>
<span id="cb43-4"><a href="week-3--working-with-files-part-2.html#cb43-4" tabindex="-1"></a>$ echo foo is $fooEFG      # doesn&#39;t work</span>
<span id="cb43-5"><a href="week-3--working-with-files-part-2.html#cb43-5" tabindex="-1"></a>foo is</span></code></pre></div>
<p>The interpreter is trying to expand a variable named <code>fooEFG</code>, which (probably)
doesn’t exist. We can avoid this problem by enclosing the variable name in
braces (<code>{</code> and <code>}</code>, sometimes called “squiggle braces”). <code>bash</code> treats the <code>#</code>
character as a comment character. Any text on a line after a <code>#</code> is ignored by
bash when evaluating the text as code.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb44-1"><a href="week-3--working-with-files-part-2.html#cb44-1" tabindex="-1"></a>$ foo=abc</span>
<span id="cb44-2"><a href="week-3--working-with-files-part-2.html#cb44-2" tabindex="-1"></a>$ echo foo is $foo</span>
<span id="cb44-3"><a href="week-3--working-with-files-part-2.html#cb44-3" tabindex="-1"></a>foo is abc</span>
<span id="cb44-4"><a href="week-3--working-with-files-part-2.html#cb44-4" tabindex="-1"></a>$ echo foo is ${foo}EFG      # now it works!</span>
<span id="cb44-5"><a href="week-3--working-with-files-part-2.html#cb44-5" tabindex="-1"></a>foo is abcEFG</span></code></pre></div>
<p>Let’s write a for loop to show us the first two lines of the fastq files we downloaded earlier. You will notice the shell prompt changes from <code>$</code> to <code>&gt;</code> and back again as we were typing in our loop. The second prompt, <code>&gt;</code>, is different to remind us that we haven’t finished typing a complete command yet. A semicolon, <code>;</code>, can be used to separate two commands written on a single line.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb45-1"><a href="week-3--working-with-files-part-2.html#cb45-1" tabindex="-1"></a>$ cd /group/rbaygrp/eve198-genomics/yourdirectory/Week2/untrimmed_fastq</span></code></pre></div>
<div class="sourceCode" id="cb46"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb46-1"><a href="week-3--working-with-files-part-2.html#cb46-1" tabindex="-1"></a>$ for filename in *.fastq</span>
<span id="cb46-2"><a href="week-3--working-with-files-part-2.html#cb46-2" tabindex="-1"></a>&gt; do</span>
<span id="cb46-3"><a href="week-3--working-with-files-part-2.html#cb46-3" tabindex="-1"></a>&gt; head -n 2 ${filename}</span>
<span id="cb46-4"><a href="week-3--working-with-files-part-2.html#cb46-4" tabindex="-1"></a>&gt; done</span></code></pre></div>
<p>The for loop begins with the formula <code>for &lt;variable&gt; in &lt;group to iterate over&gt;</code>. In this case, the word <code>filename</code> is designated
as the variable to be used over each iteration. In our case <code>SRR097977.fastq</code> and <code>SRR098026.fastq</code> will be substituted for <code>filename</code>
because they fit the pattern of ending with .fastq in the directory we’ve specified. The next line of the for loop is <code>do</code>. The next line is
the code that we want to execute. We are telling the loop to print the first two lines of each variable we iterate over. Finally, the
word <code>done</code> ends the loop.</p>
<p>After executing the loop, you should see the first two lines of both fastq files printed to the terminal. Let’s create a loop that
will save this information to a file.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb47-1"><a href="week-3--working-with-files-part-2.html#cb47-1" tabindex="-1"></a>$ for filename in *.fastq</span>
<span id="cb47-2"><a href="week-3--working-with-files-part-2.html#cb47-2" tabindex="-1"></a>&gt; do</span>
<span id="cb47-3"><a href="week-3--working-with-files-part-2.html#cb47-3" tabindex="-1"></a>&gt; head -n 2 ${filename} &gt;&gt; seq_info.txt</span>
<span id="cb47-4"><a href="week-3--working-with-files-part-2.html#cb47-4" tabindex="-1"></a>&gt; done</span></code></pre></div>
<p>When writing a loop, you will not be able to return to previous lines once you have pressed Enter. Remember that we can cancel the current command using</p>
<ul>
<li><kbd>Ctrl</kbd>+<kbd>C</kbd></li>
</ul>
<p>If you notice a mistake that is going to prevent your loop for executing correctly.</p>
<p>Note that we are using <code>&gt;&gt;</code> to append the text to our <code>seq_info.txt</code> file. If we used <code>&gt;</code>, the <code>seq_info.txt</code> file would be rewritten
every time the loop iterates, so it would only have text from the last variable used. Instead, <code>&gt;&gt;</code> adds to the end of the file.</p>
</div>
<div id="using-basename-in-for-loops" class="section level2 hasAnchor" number="4.7">
<h2><span class="header-section-number">4.7</span> Using Basename in for loops<a href="week-3--working-with-files-part-2.html#using-basename-in-for-loops" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Basename is a function in UNIX that is helpful for removing a uniform part of a name from a list of files. In this case, we will use basename to remove the <code>.fastq</code> extension from the files that we’ve been working with.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb48-1"><a href="week-3--working-with-files-part-2.html#cb48-1" tabindex="-1"></a>$ basename SRR097977.fastq .fastq</span></code></pre></div>
<p>We see that this returns just the SRR accession, and no longer has the .fastq file extension on it.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb49-1"><a href="week-3--working-with-files-part-2.html#cb49-1" tabindex="-1"></a>SRR097977</span></code></pre></div>
<p>If we try the same thing but use <code>.fasta</code> as the file extension instead, nothing happens. This is because basename only works when it exactly matches a string in the file.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb50-1"><a href="week-3--working-with-files-part-2.html#cb50-1" tabindex="-1"></a>$ basename SRR097977.fastq .fasta</span></code></pre></div>
<div class="sourceCode" id="cb51"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb51-1"><a href="week-3--working-with-files-part-2.html#cb51-1" tabindex="-1"></a>SRR097977.fastq</span></code></pre></div>
<p>Basename is really powerful when used in a for loop. It allows to access just the file prefix, which you can use to name things. Let’s try this.</p>
<p>Inside our for loop, we create a new name variable. We call the basename function inside the parenthesis, then give our variable name from the for loop, in this case <code>${filename}</code>, and finally state that <code>.fastq</code> should be removed from the file name. It’s important to note that we’re not changing the actual files, we’re creating a new variable called name. The line &gt; echo $name will print to the terminal the variable name each time the for loop runs. Because we are iterating over two files, we expect to see two lines of output.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb52-1"><a href="week-3--working-with-files-part-2.html#cb52-1" tabindex="-1"></a>$ for filename in *.fastq</span>
<span id="cb52-2"><a href="week-3--working-with-files-part-2.html#cb52-2" tabindex="-1"></a>&gt; do</span>
<span id="cb52-3"><a href="week-3--working-with-files-part-2.html#cb52-3" tabindex="-1"></a>&gt; name=$(basename ${filename} .fastq)</span>
<span id="cb52-4"><a href="week-3--working-with-files-part-2.html#cb52-4" tabindex="-1"></a>&gt; echo ${name}</span>
<span id="cb52-5"><a href="week-3--working-with-files-part-2.html#cb52-5" tabindex="-1"></a>&gt; done</span></code></pre></div>
<p>If we want to print the file prefix of all of the <code>.txt</code> files in our current directory, we would run the following code:</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb53-1"><a href="week-3--working-with-files-part-2.html#cb53-1" tabindex="-1"></a>$ for filename in *.txt</span>
<span id="cb53-2"><a href="week-3--working-with-files-part-2.html#cb53-2" tabindex="-1"></a>&gt; do</span>
<span id="cb53-3"><a href="week-3--working-with-files-part-2.html#cb53-3" tabindex="-1"></a>&gt; name=$(basename ${filename} .txt)</span>
<span id="cb53-4"><a href="week-3--working-with-files-part-2.html#cb53-4" tabindex="-1"></a>&gt; echo ${name}</span>
<span id="cb53-5"><a href="week-3--working-with-files-part-2.html#cb53-5" tabindex="-1"></a>done</span></code></pre></div>
<p>One way this is really useful is to move files. Let’s rename all of our .txt files using <code>mv</code> so that they have the years on them, which will document when we created them.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb54-1"><a href="week-3--working-with-files-part-2.html#cb54-1" tabindex="-1"></a>$ for filename in *.txt</span>
<span id="cb54-2"><a href="week-3--working-with-files-part-2.html#cb54-2" tabindex="-1"></a>&gt; do</span>
<span id="cb54-3"><a href="week-3--working-with-files-part-2.html#cb54-3" tabindex="-1"></a>&gt; name=$(basename ${filename} .txt)</span>
<span id="cb54-4"><a href="week-3--working-with-files-part-2.html#cb54-4" tabindex="-1"></a>&gt; mv ${filename}  ${name}_2019.txt</span>
<span id="cb54-5"><a href="week-3--working-with-files-part-2.html#cb54-5" tabindex="-1"></a>&gt; done</span></code></pre></div>
<p>What if we put the wrong year and actually want to remove the 2019 from all of our .txt files?</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb55-1"><a href="week-3--working-with-files-part-2.html#cb55-1" tabindex="-1"></a> $ for filename in *_2019.txt</span>
<span id="cb55-2"><a href="week-3--working-with-files-part-2.html#cb55-2" tabindex="-1"></a>&gt; do</span>
<span id="cb55-3"><a href="week-3--working-with-files-part-2.html#cb55-3" tabindex="-1"></a>&gt; name=$(basename ${filename} _2019.txt)</span>
<span id="cb55-4"><a href="week-3--working-with-files-part-2.html#cb55-4" tabindex="-1"></a>&gt; mv ${filename} ${name}.txt</span>
<span id="cb55-5"><a href="week-3--working-with-files-part-2.html#cb55-5" tabindex="-1"></a>&gt; done</span></code></pre></div>
<p>Now write your own for loop it to have <code>_2025</code> at the end of each of our .txt files!</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb56-1"><a href="week-3--working-with-files-part-2.html#cb56-1" tabindex="-1"></a>$ for filename in *.txt</span>
<span id="cb56-2"><a href="week-3--working-with-files-part-2.html#cb56-2" tabindex="-1"></a>&gt; do</span>
<span id="cb56-3"><a href="week-3--working-with-files-part-2.html#cb56-3" tabindex="-1"></a>&gt; name=$(basename ${filename} .txt)</span>
<span id="cb56-4"><a href="week-3--working-with-files-part-2.html#cb56-4" tabindex="-1"></a>&gt; mv ${filename}  ${name}_2025.txt</span>
<span id="cb56-5"><a href="week-3--working-with-files-part-2.html#cb56-5" tabindex="-1"></a>&gt; done</span></code></pre></div>
</div>
<div id="writing-scripts-and-working-with-data" class="section level2 hasAnchor" number="4.8">
<h2><span class="header-section-number">4.8</span> Writing Scripts and Working with Data<a href="week-3--working-with-files-part-2.html#writing-scripts-and-working-with-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We’ve been able to do a lot of work with files that already exist, but what if we want to write our own files? We’re not going to type in a FASTA file, but we’ll see as we go through other tutorials, there are a lot of reasons we’ll want to write a file, or edit an existing file.</p>
<p>To add text to files, we’re going to use a text editor called Nano. We’re going to create a file to take notes about what we’ve been doing with the data files in <code>~/group/rbaygrp/eve198-genomics/yourdirectory/Week2/untrimmed_fastq</code>.</p>
<p>This is good practice when working in bioinformatics. We can create a file called <code>README.txt</code> that describes the data files in the directory or documents how the files in that directory were generated. As the name suggests, it’s a file that we or others should read to understand the information in that directory.</p>
<p>Let’s change our working directory to <code>~/group/rbaygrp/eve198-genomics/yourdirectory/Week2/untrimmed_fastq</code> using <code>cd</code>,
then run <code>nano</code> to create a file called <code>README.txt</code>:</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb57-1"><a href="week-3--working-with-files-part-2.html#cb57-1" tabindex="-1"></a>$ cd ~/group/rbaygrp/eve198-genomics/yourdirectory/Week2/untrimmed_fastq</span>
<span id="cb57-2"><a href="week-3--working-with-files-part-2.html#cb57-2" tabindex="-1"></a>$ nano README.txt</span></code></pre></div>
<p>You should see something like this:</p>
<div class="float">
<img src="figs/nano201711%20copy.png" alt="nano201711.png" />
<div class="figcaption">nano201711.png</div>
</div>
<p>The text at the bottom of the screen shows the keyboard shortcuts for performing various tasks in <code>nano</code>. We will talk more about how to interpret this information soon.</p>
<details>
<summary>
Which Editor?
</summary>
<p>
<blockquote>
<h3 id="which-editor" class="hasAnchor">Which Editor?<a href="week-3--working-with-files-part-2.html#which-editor" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>When we say, “<code>nano</code> is a text editor,” we really do mean “text”: it can
only work with plain character data, not tables, images, or any other
human-friendly media. We use it in examples because it is one of the
least complex text editors. However, because of this trait, it may
not be powerful enough or flexible enough for the work you need to do
after this workshop. On Unix systems (such as Linux and Mac OS X),
many programmers use <a href="http://www.gnu.org/software/emacs/">Emacs</a> or
<a href="http://www.vim.org/">Vim</a> (both of which require more time to learn),
or a graphical editor such as
<a href="http://projects.gnome.org/gedit/">Gedit</a>. On Windows, you may wish to
use <a href="http://notepad-plus-plus.org/">Notepad++</a>. Windows also has a built-in
editor called <code>notepad</code> that can be run from the command line in the same
way as <code>nano</code> for the purposes of this lesson.</p>
<p>No matter what editor you use, you will need to know where it searches
for and saves files. If you start it from the shell, it will (probably)
use your current working directory as its default location. If you use
your computer’s start menu, it may want to save files in your desktop or
documents directory instead. You can change this by navigating to
another directory the first time you “Save As…”</p>
</blockquote>
</p>
</details>
<p> </p>
<p>Let’s type in a few lines of text. Describe what the files in this
directory are or what you’ve been doing with them.
Once we’re happy with our text, we can press <kbd>Ctrl</kbd>-<kbd>O</kbd> (press the <kbd>Ctrl</kbd> or <kbd>Control</kbd> key and, while
holding it down, press the <kbd>O</kbd> key) to write our data to disk. You’ll be asked what file we want to save this to:
press <kbd>Return</kbd> to accept the suggested default of <code>README.txt</code>.</p>
<p>Once our file is saved, we can use <kbd>Ctrl</kbd>-<kbd>X</kbd> to quit the editor and
return to the shell.</p>
<blockquote>
<h3 id="control-ctrl-or-key" class="hasAnchor">Control, Ctrl, or ^ Key<a href="week-3--working-with-files-part-2.html#control-ctrl-or-key" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The Control key is also called the “Ctrl” key. There are various ways
in which using the Control key may be described. For example, you may
see an instruction to press the <kbd>Ctrl</kbd> key and, while holding it down,
press the <kbd>X</kbd> key, described as any of:</p>
<ul>
<li><code>Control-X</code></li>
<li><code>Control+X</code></li>
<li><code>Ctrl-X</code></li>
<li><code>Ctrl+X</code></li>
<li><code>^X</code></li>
<li><code>C-x</code></li>
</ul>
<p>In <code>nano</code>, along the bottom of the screen you’ll see <code>^G Get Help ^O WriteOut</code>.
This means that you can use <kbd>Ctrl</kbd>-<kbd>G</kbd> to get help and <kbd>Ctrl</kbd>-<kbd>O</kbd> to save your
file.</p>
</blockquote>
<p>Now you’ve written a file. You can take a look at it with <code>less</code> or <code>cat</code>, or open it up again and edit it with <code>nano</code>.</p>
</div>
<div id="writing-scripts-bash" class="section level2 hasAnchor" number="4.9">
<h2><span class="header-section-number">4.9</span> Writing scripts: Bash<a href="week-3--working-with-files-part-2.html#writing-scripts-bash" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A really powerful thing about the command line is that you can write scripts. Scripts let you save commands to run them and also lets you put multiple commands together. Though writing scripts may require an additional time investment initially, this can save you time as you run them repeatedly. Scripts can also address the challenge of reproducibility: if you need to repeat an analysis, you retain a record of your command history within the script.</p>
<p>One thing we will commonly want to do with sequencing results is pull out bad reads and write them to a file to see if we can figure out what’s going on with them. We’re going to look for reads with long sequences of N’s like we did before, but now we’re going to write a script, so we can run it each time we get new sequences, rather than type the code in by hand each time.</p>
<p>We’re going to create a new file to put this command in. We’ll call it <code>bad-reads-script.sh</code>. The <code>sh</code> isn’t required, but using that extension tells us that it’s a shell script.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb58-1"><a href="week-3--working-with-files-part-2.html#cb58-1" tabindex="-1"></a>$ nano bad-reads-script.sh</span></code></pre></div>
<p>Bad reads have a lot of N’s, so we’re going to look for <code>NNNNNNNNNN</code> with <code>grep</code>. We want the whole FASTQ record, so we’re also going to get the one line above the sequence and the two lines below. We also want to look in all the files that end with <code>.fastq</code>, so we’re going to use the <code>*</code> wildcard.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb59-1"><a href="week-3--working-with-files-part-2.html#cb59-1" tabindex="-1"></a>grep -B1 -A2 -h NNNNNNNNNN *.fastq | grep -v &#39;^--&#39; &gt; scripted_bad_reads.txt</span></code></pre></div>
<blockquote>
<h3 id="custom-grep-control-1" class="hasAnchor">Custom <code>grep</code> control<a href="week-3--working-with-files-part-2.html#custom-grep-control-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We introduced the <code>-v</code> option in <a href="http://www.datacarpentry.org/shell-genomics/04-redirection/">the previous lesson</a>, now we
are using <code>-h</code> to “Suppress the prefixing of file names on output” according to the documentation shown by <code>man grep</code>.</p>
</blockquote>
<p>Type your <code>grep</code> command into the file and save it as before. Be careful that you did not add the <code>$</code> at the beginning of the line.</p>
<p>Now comes the neat part. We can run this script. Type:</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb60-1"><a href="week-3--working-with-files-part-2.html#cb60-1" tabindex="-1"></a>$ bash bad-reads-script.sh</span></code></pre></div>
<p>It will look like nothing happened, but now if you look at <code>scripted_bad_reads.txt</code>, you can see that there are now reads in the file.</p>
<p>If we want our ‘bad-reads-script.sh’ to tell us when it is done running, we can add this line at the end:</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb61-1"><a href="week-3--working-with-files-part-2.html#cb61-1" tabindex="-1"></a>grep echo &quot;Script finished!&quot;</span></code></pre></div>
<p>If you run the bash bad-reads-script.sh command again and it will print out “Script finished!”</p>
</div>
<div id="writing-scripts-slurm" class="section level2 hasAnchor" number="4.10">
<h2><span class="header-section-number">4.10</span> Writing scripts: Slurm<a href="week-3--working-with-files-part-2.html#writing-scripts-slurm" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can also write scripts using a resource manager called “Slurm” (It is a reference to the show Futurama where there is a soda called slurm). Slurm can be really helpful for writing scripts and submitting jobs since you can schedule and run larger jobs. I use this personally in my own research. The set up is pretty similar. First let’s create a simple script:</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb62-1"><a href="week-3--working-with-files-part-2.html#cb62-1" tabindex="-1"></a>nano HelloWorld.sh</span></code></pre></div>
<p>For a slurm script we need more information in the header of our script so that it knows what account to run on and the amount of time you want to give to your job:</p>
<ul>
<li>All scripts start with <code>#!/bin/bash</code></li>
<li>The next line is your email address so you can get an email when your job finishes</li>
<li>Then add information on if you want an email for all notifications (began, ended, failed, etc)</li>
<li>Next is the name of the job, or job ID</li>
<li>The name of your error log file</li>
<li>The name of your output log file (usually the same name as your error)</li>
<li>The number of processors per task</li>
<li>The minimum number of nodes you need for your job. Small jobs don’t need a lot of nodes/memory</li>
<li>The amount of memory for your job</li>
<li>The requested time for your job. Some jobs take a really long time, I had one run for several months! For this class you only need to put in 90 minutes.</li>
<li>The partition to submit to, high2 is the one we will be using</li>
</ul>
<p>After those headers you write out your script!</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb63-1"><a href="week-3--working-with-files-part-2.html#cb63-1" tabindex="-1"></a>#!/bin/bash</span>
<span id="cb63-2"><a href="week-3--working-with-files-part-2.html#cb63-2" tabindex="-1"></a>#SBATCH --mail-user=mlarmstrong@ucdavis.edu         # YOUR EMAIL ADDRESS</span>
<span id="cb63-3"><a href="week-3--working-with-files-part-2.html#cb63-3" tabindex="-1"></a>#SBATCH --account=rbay-eve198-genomics-grp    #we need this to specify what account we are using</span>
<span id="cb63-4"><a href="week-3--working-with-files-part-2.html#cb63-4" tabindex="-1"></a>#SBATCH --mail-type=ALL                         # NOTIFICATIONS OF SLURM JOB STATUS - ALL, NONE, BEGIN, END, FAIL, REQUEUE</span>
<span id="cb63-5"><a href="week-3--working-with-files-part-2.html#cb63-5" tabindex="-1"></a>#SBATCH -J HelloWorld                           # JOB ID</span>
<span id="cb63-6"><a href="week-3--working-with-files-part-2.html#cb63-6" tabindex="-1"></a>#SBATCH -e HelloWorld.j%j.err                   # STANDARD ERROR FILE TO WRITE TO</span>
<span id="cb63-7"><a href="week-3--working-with-files-part-2.html#cb63-7" tabindex="-1"></a>#SBATCH -o HelloWorld.j%j.out                   # STANDARD OUTPUT FILE TO WRITE TO</span>
<span id="cb63-8"><a href="week-3--working-with-files-part-2.html#cb63-8" tabindex="-1"></a>#SBATCH -c 1                                    # NUMBER OF PROCESSORS PER TASK</span>
<span id="cb63-9"><a href="week-3--working-with-files-part-2.html#cb63-9" tabindex="-1"></a>#SBATCH --ntasks=1                              # MINIMUM NUMBER OF NODES TO ALLOCATE TO JOB</span>
<span id="cb63-10"><a href="week-3--working-with-files-part-2.html#cb63-10" tabindex="-1"></a>#SBATCH --mem=1Gb                               # MEMORY POOL TO ALL CORES</span>
<span id="cb63-11"><a href="week-3--working-with-files-part-2.html#cb63-11" tabindex="-1"></a>#SBATCH --time=00-00:11:00                      # REQUESTED WALL TIME</span>
<span id="cb63-12"><a href="week-3--working-with-files-part-2.html#cb63-12" tabindex="-1"></a>#SBATCH -p high2                               # PARTITION TO SUBMIT TO</span>
<span id="cb63-13"><a href="week-3--working-with-files-part-2.html#cb63-13" tabindex="-1"></a></span>
<span id="cb63-14"><a href="week-3--working-with-files-part-2.html#cb63-14" tabindex="-1"></a>echo &quot;Hello World!&quot; &gt; hello_world_output.txt</span></code></pre></div>
<p>Then when you are ready to run your job you type</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb64-1"><a href="week-3--working-with-files-part-2.html#cb64-1" tabindex="-1"></a>sbatch HelloWorld.sh</span></code></pre></div>
<p>Want to check the status of your job? Type <code>squeue -u</code>yourusername<code>Once your job is done, type</code>ls<code>. There should be a new text file in your directory called</code>hello_world_output.txt`. What does it say?</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb65-1"><a href="week-3--working-with-files-part-2.html#cb65-1" tabindex="-1"></a>cat hello_world_output.txt</span></code></pre></div>
<p>More information on slurm scripts in on the hpc webpage: <a href="https://hpc.ucdavis.edu/faq/slurm" class="uri">https://hpc.ucdavis.edu/faq/slurm</a></p>
</div>
<div id="group-work-activity--writing-your-own-script" class="section level2 hasAnchor" number="4.11">
<h2><span class="header-section-number">4.11</span> Group Work Activity- Writing your own script<a href="week-3--working-with-files-part-2.html#group-work-activity--writing-your-own-script" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>modified from <a href="https://www.cyberciti.biz/faq/hello-world-bash-shell-script/" class="uri">https://www.cyberciti.biz/faq/hello-world-bash-shell-script/</a></p>
<p>Now let’s work in small groups to write our own slurm scripts. The script should be called “hello.sh” and you want it to write a text file that says “Hello World! Isn’t genomics so fun?”</p>
<p>Remember your script should start with <code>#!/bin/bash</code>. Submit a copy of your script on canvas under the ‘Assignments’ tab for ‘Week 3: Writing your own script’.</p>
</div>
<div id="key-points-2" class="section level2 hasAnchor" number="4.12">
<h2><span class="header-section-number">4.12</span> Key Points<a href="week-3--working-with-files-part-2.html#key-points-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p>The commands <code>cp</code>, <code>mv</code>, and <code>mkdir</code> are useful for manipulating existing files and creating new directories.</p></li>
<li><p>You can view file permissions using <code>ls -l</code> and change permissions using <code>chmod</code>.</p></li>
<li><p><code>grep</code> is a powerful search tool with many options for customization.</p></li>
<li><p><code>&gt;</code>, <code>&gt;&gt;</code>, and <code>|</code> are different ways of redirecting output.</p></li>
<li><p><code>command &gt; file</code> redirects a command’s output to a file whle <code>command &gt;&gt; file</code> redirects a command’s output to a file without overwriting the existing contents of the file.</p></li>
<li><p><code>command_1 | command_2</code> redirects the output of the first command as input to the second command.</p></li>
<li><p><code>for</code> loops are used for iteration.</p></li>
<li><p><code>basename</code> gets rid of repetitive parts of names.</p></li>
<li><p>Scripts are a collection of commands executed together.</p></li>
</ul>
<details>
<summary>
<span style="color: orange;">Class Exercise Solutions</span>
</summary>
<p>
<blockquote>
<blockquote>
<h2 id="exercise-1-solution" class="hasAnchor">Exercise 1: Solution<a href="week-3--working-with-files-part-2.html#exercise-1-solution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><code>rm -r backup</code><br />
</li>
<li><code>cp SRR098026.fastq SRR098026-backup.fastq</code> and <code>cp SRR097977.fastq SRR097977-backup.fastq</code><br />
</li>
<li><code>mkdir backup</code> and <code>mv *-backup.fastq backup</code></li>
<li><code>chmod -w backup/*-backup.fastq</code><br />
It’s always a good idea to check your work with <code>ls -l backup</code>. You should see something like:</li>
</ol>
<div class="sourceCode" id="cb66"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb66-1"><a href="week-3--working-with-files-part-2.html#cb66-1" tabindex="-1"></a>-r--r--r-- 1 dcuser dcuser 47552 Nov 15 23:06 SRR097977-backup.fastq</span>
<span id="cb66-2"><a href="week-3--working-with-files-part-2.html#cb66-2" tabindex="-1"></a>-r--r--r-- 1 dcuser dcuser 43332 Nov 15 23:06 SRR098026-backup.fastq</span></code></pre></div>
<h3 id="exercise-2-solution" class="hasAnchor">Exercise 2: Solution<a href="week-3--working-with-files-part-2.html#exercise-2-solution" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><code>grep -B1 GNATNACCACTTCC SRR098026.fastq</code></p>
<pre><code>@SRR098026.245 HWUSI-EAS1599_1:2:1:2:801 length=35
GNATNACCACTTCCAGTGCTGANNNNNNNGGGATG</code></pre></li>
<li><p><code>grep -B1 AAGTT *.fastq</code></p>
<pre><code>SRR097977.fastq-@SRR097977.11 209DTAAXX_Lenski2_1_7:8:3:247:351 length=36
SRR097977.fastq:GATTGCTTTAATGAAAAAGTCATATAAGTTGCCATG
--
SRR097977.fastq-@SRR097977.67 209DTAAXX_Lenski2_1_7:8:3:544:566 length=36
SRR097977.fastq:TTGTCCACGCTTTTCTATGTAAAGTTTATTTGCTTT
--
SRR097977.fastq-@SRR097977.68 209DTAAXX_Lenski2_1_7:8:3:724:110 length=36
SRR097977.fastq:TGAAGCCTGCTTTTTTATACTAAGTTTGCATTATAA
--
SRR097977.fastq-@SRR097977.80 209DTAAXX_Lenski2_1_7:8:3:258:281 length=36
SRR097977.fastq:GTGGCGCTGCTGCATAAGTTGGGTTATCAGGTCGTT
--
SRR097977.fastq-@SRR097977.92 209DTAAXX_Lenski2_1_7:8:3:353:318 length=36
SRR097977.fastq:GGCAAAATGGTCCTCCAGCCAGGCCAGAAGCAAGTT
--
SRR097977.fastq-@SRR097977.139 209DTAAXX_Lenski2_1_7:8:3:703:655 length=36
SRR097977.fastq:TTTATTTGTAAAGTTTTGTTGAAATAAGGGTTGTAA
--
SRR097977.fastq-@SRR097977.238 209DTAAXX_Lenski2_1_7:8:3:592:919 length=36
SRR097977.fastq:TTCTTACCATCCTGAAGTTTTTTCATCTTCCCTGAT
--
SRR098026.fastq-@SRR098026.158 HWUSI-EAS1599_1:2:1:1:1505 length=35
SRR098026.fastq:GNNNNNNNNCAAAGTTGATCNNNNNNNNNTGTGCG</code></pre></li>
</ol>
</blockquote>
</blockquote>
</div>
<div id="additional-information" class="section level2 hasAnchor" number="4.13">
<h2><span class="header-section-number">4.13</span> Additional Information<a href="week-3--working-with-files-part-2.html#additional-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix" class="section level3 hasAnchor" number="4.13.1">
<h3><span class="header-section-number">4.13.1</span> Uploading and Downloading Data to your Virtual Machine with scp - UNIX<a href="week-3--working-with-files-part-2.html#uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This section is for your general knowledge.</p>
<p><code>scp</code> stands for ‘secure copy protocol’, and is a widely used UNIX tool for moving files
between computers. The simplest way to use <code>scp</code> is to run it in your local terminal,
and use it to copy a single file:</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb69-1"><a href="week-3--working-with-files-part-2.html#cb69-1" tabindex="-1"></a>scp <span class="dt">&lt;</span><span class="kw">file</span><span class="ot"> I want to move</span><span class="dt">&gt;</span> <span class="dt">&lt;</span><span class="kw">where</span><span class="ot"> I want to move it</span><span class="dt">&gt;</span></span></code></pre></div>
<p>In terminal,you can use the <code>scp</code> command to upload a file (e.g. local_file.txt) to the cluster home directory:</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb70-1"><a href="week-3--working-with-files-part-2.html#cb70-1" tabindex="-1"></a>$  scp local_file.txt UserName@cluster.address:/scratch/</span></code></pre></div>
<p>If you wanted to download data from your virtual machine, we would put the location of the folder within the virtual machine in the location of <file I want to move></p>
<div class="sourceCode" id="cb71"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb71-1"><a href="week-3--working-with-files-part-2.html#cb71-1" tabindex="-1"></a>$  scp UserName@cluster.address:/scratch/VirtualMachine_file.txt /group/rbaygrp/eve198-genomics/`yourdirectory`/Week2/untrimmed_fastq/</span></code></pre></div>
</div>
</div>
<div id="more-ways-to-work-with-files" class="section level2 hasAnchor" number="4.14">
<h2><span class="header-section-number">4.14</span> More ways to work with files<a href="week-3--working-with-files-part-2.html#more-ways-to-work-with-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="awk" class="section level3 hasAnchor" number="4.14.1">
<h3><span class="header-section-number">4.14.1</span> Awk<a href="week-3--working-with-files-part-2.html#awk" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><code>Awk</code> is a fast and versatile pattern matching programming language. <code>Awk</code> can do the same tasks that <code>sed</code>, <code>grep</code>, <code>cat</code>, and <code>wc</code>; and then it can do a lot more <a href="https://www.gnu.org/software/gawk/manual/gawk.html" class="uri">https://www.gnu.org/software/gawk/manual/gawk.html</a>. This program deserves a full class to go into details, so instead we just have this section to make you aware that the program exists.</p>
<p>Let’s see how awk can behave like <code>wc</code>.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb72-1"><a href="week-3--working-with-files-part-2.html#cb72-1" tabindex="-1"></a>$ cd /group/rbaygrp/eve198-genomics/`yourdirectory`/Week2</span></code></pre></div>
<div class="sourceCode" id="cb73"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb73-1"><a href="week-3--working-with-files-part-2.html#cb73-1" tabindex="-1"></a>$ ls TableS2_QTL_Bay_2017.txt  sra_metadata  untrimmed_fastq</span></code></pre></div>
<p>This table is from the Bay et al. 2017 publication <code>TableS2_QTL_Bay_2017.txt</code> and we will use it as our example file for this section.</p>
<p>We can look inside the file by using <code>cat</code> or <code>awk</code></p>
<div class="sourceCode" id="cb74"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb74-1"><a href="week-3--working-with-files-part-2.html#cb74-1" tabindex="-1"></a>$ awk &#39;{print $0}&#39; TableS2_QTL_Bay_2017.txt</span></code></pre></div>
<p>The instructions are enclosed in single quotes</p>
<p>This command has the same output of “cat”: it prints each line from the example file TableS2_QTL_Bay_2017.txt</p>
<p>The structure of the instruction is the following:
- curly braces surround the set of instructions
- print is the instruction that sends its arguments to the terminal
- $0 is a variable, it means “the content of the current line”</p>
<p>As you can see, the file contains a table.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb75-1"><a href="week-3--working-with-files-part-2.html#cb75-1" tabindex="-1"></a>Trait   n   LOD Chr Position (cM)   Nearest SNP </span>
<span id="cb75-2"><a href="week-3--working-with-files-part-2.html#cb75-2" tabindex="-1"></a>mate choice 200 4.5 14  22.43   chrXIV:1713227 </span>
<span id="cb75-3"><a href="week-3--working-with-files-part-2.html#cb75-3" tabindex="-1"></a>mate choice     200 4.61    21  8   chrXXI:9373717 </span>
<span id="cb75-4"><a href="week-3--working-with-files-part-2.html#cb75-4" tabindex="-1"></a>discriminant function   200 4.83    12  17  chrXII:7504339 </span>
<span id="cb75-5"><a href="week-3--working-with-files-part-2.html#cb75-5" tabindex="-1"></a>discriminant function   200 4.23    14  8.1 chrXIV:4632223 </span>
<span id="cb75-6"><a href="week-3--working-with-files-part-2.html#cb75-6" tabindex="-1"></a>PC2 200 4.04    4   30.76   chrIV:11367975 </span>
<span id="cb75-7"><a href="week-3--working-with-files-part-2.html#cb75-7" tabindex="-1"></a>PC2 200 6.67    7   47  chrVII:26448674 </span>
<span id="cb75-8"><a href="week-3--working-with-files-part-2.html#cb75-8" tabindex="-1"></a>centroid size   200 6.97    9   47.8    chrIX:19745222 </span>
<span id="cb75-9"><a href="week-3--working-with-files-part-2.html#cb75-9" tabindex="-1"></a>x2* 200 3.93    7   60  chrUn:29400087 </span>
<span id="cb75-10"><a href="week-3--working-with-files-part-2.html#cb75-10" tabindex="-1"></a>y2* 200 9.99    4   32  chrIV:11367975 </span>
<span id="cb75-11"><a href="week-3--working-with-files-part-2.html#cb75-11" tabindex="-1"></a>x3  200 4.45    1   32.3    chrI:15145305 </span>
<span id="cb75-12"><a href="week-3--working-with-files-part-2.html#cb75-12" tabindex="-1"></a>x4  200 5.13    16  30.9    chrXVI:12111717 </span>
<span id="cb75-13"><a href="week-3--working-with-files-part-2.html#cb75-13" tabindex="-1"></a>x5* 200 4.54    15  6   chrXV:505537 </span>
<span id="cb75-14"><a href="week-3--working-with-files-part-2.html#cb75-14" tabindex="-1"></a>y5  200 4.21    4   24.9    chrIV:15721538 </span>
<span id="cb75-15"><a href="week-3--working-with-files-part-2.html#cb75-15" tabindex="-1"></a>x6  200 3.96    16  29.5    chrXVI:13588796 </span>
<span id="cb75-16"><a href="week-3--working-with-files-part-2.html#cb75-16" tabindex="-1"></a>y6* 200 4.14    9   30.2    chrIX:18942598 </span>
<span id="cb75-17"><a href="week-3--working-with-files-part-2.html#cb75-17" tabindex="-1"></a>y15*    200 5.3 2   27  chrII:19324477 </span>
<span id="cb75-18"><a href="week-3--working-with-files-part-2.html#cb75-18" tabindex="-1"></a>x16 200 5.49    7   60  chrUn:29400087 </span>
<span id="cb75-19"><a href="week-3--working-with-files-part-2.html#cb75-19" tabindex="-1"></a>x17     200 4.92    1   32.8    chrI:14261764 </span>
<span id="cb75-20"><a href="week-3--working-with-files-part-2.html#cb75-20" tabindex="-1"></a>Table S2. Significant QTL loci for mate choice and morphology</span></code></pre></div>
<p>Now let’s use <code>awk</code> to count the lines of a file, similarly to what <code>wc -l</code> would do.</p>
<p>As you probably remember, -l is an option that asks for the number of lines only.</p>
<p>However, wc counts the number of newlines in the file, if the last line does
not contain a carriage return (i.e. there is no emptyline at the end of the file),
the result is going be the actual number of lines minus one.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb76-1"><a href="week-3--working-with-files-part-2.html#cb76-1" tabindex="-1"></a>$ wc -l TableS2_QTL_Bay_2017.txt</span></code></pre></div>
<div class="sourceCode" id="cb77"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb77-1"><a href="week-3--working-with-files-part-2.html#cb77-1" tabindex="-1"></a>19 TableS2_QTL_Bay_2017.txt</span></code></pre></div>
<p>A workaround is to use <code>awk</code>. <code>Awk</code> is command line program that takes as input a set
of instructions and one or more files. The instructions are executed on each line
of the input file(s).</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb78-1"><a href="week-3--working-with-files-part-2.html#cb78-1" tabindex="-1"></a>$ awk &#39;{print NR;}&#39; TableS2_QTL_Bay_2017.txt | tail -1</span></code></pre></div>
<p><code>Awk</code> can also search within a file like <code>grep</code> can. Let’s see if there are any significant QTL loci in the chromosome “chrXIV”</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb79-1"><a href="week-3--working-with-files-part-2.html#cb79-1" tabindex="-1"></a>$ awk &#39;/chrXIV/&#39; TableS2_QTL_Bay_2017.txt</span></code></pre></div>
<p>This chromosome had two significant QTL Loci for mate choice and morphology.</p>
<p>When to use awk?</p>
<ul>
<li>for search and replacement of large files (it’s fast!)</li>
<li>when manipulating multiple large files</li>
</ul>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="week-2--working-with-files.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="week-4--what-is-a-genetic-variant.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
