<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 3 Week 2- Working with Files | 02-Week2.Rmd</title>
  <meta name="description" content=" 3 Week 2- Working with Files | 02-Week2.Rmd" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content=" 3 Week 2- Working with Files | 02-Week2.Rmd" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 3 Week 2- Working with Files | 02-Week2.Rmd" />
  
  
  

<meta name="author" content="M. Armstrong" />


<meta name="date" content="2025-03-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="week-1--welcome.html"/>
<link rel="next" href="week-3--working-with-files-part-2.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to Genomics in Natural Populations</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro to Genomics in Natural Populations</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#course-learning-objectives"><i class="fa fa-check"></i><b>1.1</b> Course Learning Objectives</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#meet-your-instructors"><i class="fa fa-check"></i><b>1.2</b> Meet your instructors!</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#getting-your-farm-account-set-up-class-specific"><i class="fa fa-check"></i><b>1.3</b> Getting your Farm Account set up (class specific)</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="week-1--welcome.html"><a href="week-1--welcome.html"><i class="fa fa-check"></i><b>2</b> Week 1- Welcome!</a>
<ul>
<li class="chapter" data-level="2.1" data-path="week-1--welcome.html"><a href="week-1--welcome.html#main-objectives"><i class="fa fa-check"></i><b>2.1</b> Main Objectives</a></li>
<li class="chapter" data-level="2.2" data-path="week-1--welcome.html"><a href="week-1--welcome.html#how-to-access-the-shell-via-farm-ondemand"><i class="fa fa-check"></i><b>2.2</b> How to access the shell via Farm OnDemand</a></li>
<li class="chapter" data-level="2.3" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-your-file-system"><i class="fa fa-check"></i><b>2.3</b> Navigating your file system</a></li>
<li class="chapter" data-level="2.4" data-path="week-1--welcome.html"><a href="week-1--welcome.html#shortcut-tab-completion"><i class="fa fa-check"></i><b>2.4</b> Shortcut: Tab Completion</a></li>
<li class="chapter" data-level="2.5" data-path="week-1--welcome.html"><a href="week-1--welcome.html#creating-our-own-directories"><i class="fa fa-check"></i><b>2.5</b> Creating our own directories</a></li>
<li class="chapter" data-level="2.6" data-path="week-1--welcome.html"><a href="week-1--welcome.html#full-vs-relative-paths"><i class="fa fa-check"></i><b>2.6</b> Full vs Relative Paths</a></li>
<li class="chapter" data-level="2.7" data-path="week-1--welcome.html"><a href="week-1--welcome.html#examining-the-contents-of-other-directories"><i class="fa fa-check"></i><b>2.7</b> Examining the contents of other directories</a></li>
<li class="chapter" data-level="2.8" data-path="week-1--welcome.html"><a href="week-1--welcome.html#group-work-activity--treasure-hunt"><i class="fa fa-check"></i><b>2.8</b> Group Work Activity- Treasure Hunt!</a></li>
<li class="chapter" data-level="2.9" data-path="week-1--welcome.html"><a href="week-1--welcome.html#key-points"><i class="fa fa-check"></i><b>2.9</b> Key Points</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html"><i class="fa fa-check"></i><b>3</b> Week 2- Working with Files</a>
<ul>
<li class="chapter" data-level="3.1" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#main-objectives-1"><i class="fa fa-check"></i><b>3.1</b> Main Objectives:</a></li>
<li class="chapter" data-level="3.2" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#navigational-shortcuts"><i class="fa fa-check"></i><b>3.2</b> Navigational shortcuts</a></li>
<li class="chapter" data-level="3.3" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#details-on-the-fastq-format"><i class="fa fa-check"></i><b>3.3</b> Details on the FASTQ format</a></li>
<li class="chapter" data-level="3.4" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#our-data-set-fastq-files"><i class="fa fa-check"></i><b>3.4</b> Our data set: FASTQ files</a></li>
<li class="chapter" data-level="3.5" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#wildcards"><i class="fa fa-check"></i><b>3.5</b> Wildcards</a></li>
<li class="chapter" data-level="3.6" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#command-history"><i class="fa fa-check"></i><b>3.6</b> Command History</a></li>
<li class="chapter" data-level="3.7" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#examining-files"><i class="fa fa-check"></i><b>3.7</b> Examining Files</a></li>
<li class="chapter" data-level="3.8" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#group-work-activity--examining-a-fastq-file"><i class="fa fa-check"></i><b>3.8</b> Group Work Activity- Examining a Fastq File</a></li>
<li class="chapter" data-level="3.9" data-path="week-2--working-with-files.html"><a href="week-2--working-with-files.html#key-points-1"><i class="fa fa-check"></i><b>3.9</b> Key Points</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html"><i class="fa fa-check"></i><b>4</b> Week 3- Working with Files Part 2</a>
<ul>
<li class="chapter" data-level="4.1" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#main-objectives-2"><i class="fa fa-check"></i><b>4.1</b> Main Objectives</a></li>
<li class="chapter" data-level="4.2" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#creating-moving-copying-and-removing"><i class="fa fa-check"></i><b>4.2</b> Creating, moving, copying, and removing</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#copying-files"><i class="fa fa-check"></i><b>4.2.1</b> Copying Files</a></li>
<li class="chapter" data-level="4.2.2" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#creating-directories"><i class="fa fa-check"></i><b>4.2.2</b> Creating Directories</a></li>
<li class="chapter" data-level="4.2.3" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#moving-renaming"><i class="fa fa-check"></i><b>4.2.3</b> Moving / Renaming</a></li>
<li class="chapter" data-level="4.2.4" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#file-permissions"><i class="fa fa-check"></i><b>4.2.4</b> File Permissions</a></li>
<li class="chapter" data-level="4.2.5" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#removing"><i class="fa fa-check"></i><b>4.2.5</b> Removing</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#searching-files"><i class="fa fa-check"></i><b>4.3</b> Searching files</a></li>
<li class="chapter" data-level="4.4" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#redirecting-output"><i class="fa fa-check"></i><b>4.4</b> Redirecting output</a></li>
<li class="chapter" data-level="4.5" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#file-extensions"><i class="fa fa-check"></i><b>4.5</b> File extensions</a></li>
<li class="chapter" data-level="4.6" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#writing-for-loops"><i class="fa fa-check"></i><b>4.6</b> Writing for loops</a></li>
<li class="chapter" data-level="4.7" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#using-basename-in-for-loops"><i class="fa fa-check"></i><b>4.7</b> Using Basename in for loops</a></li>
<li class="chapter" data-level="4.8" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#moving-and-downloading-data"><i class="fa fa-check"></i><b>4.8</b> Moving and Downloading Data</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#getting-data-from-the-cloud"><i class="fa fa-check"></i><b>4.8.1</b> Getting data from the cloud</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#writing-scripts-and-working-with-data"><i class="fa fa-check"></i><b>4.9</b> Writing Scripts and Working with Data</a></li>
<li class="chapter" data-level="4.10" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#writing-scripts-bash"><i class="fa fa-check"></i><b>4.10</b> Writing scripts: Bash</a></li>
<li class="chapter" data-level="4.11" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#writing-scripts-slurm"><i class="fa fa-check"></i><b>4.11</b> Writing scripts: Slurm</a></li>
<li class="chapter" data-level="4.12" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#group-work-activity--writing-your-own-script"><i class="fa fa-check"></i><b>4.12</b> Group Work Activity- Writing your own script</a></li>
<li class="chapter" data-level="4.13" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#key-points-2"><i class="fa fa-check"></i><b>4.13</b> Key Points</a></li>
<li class="chapter" data-level="4.14" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#additional-information-on-working-with-files"><i class="fa fa-check"></i><b>4.14</b> Additional Information on working with files:</a>
<ul>
<li class="chapter" data-level="4.14.1" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix"><i class="fa fa-check"></i><b>4.14.1</b> Uploading and Downloading Data to your Virtual Machine with scp - UNIX</a></li>
<li class="chapter" data-level="4.14.2" data-path="week-3--working-with-files-part-2.html"><a href="week-3--working-with-files-part-2.html#awk"><i class="fa fa-check"></i><b>4.14.2</b> Awk</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html"><i class="fa fa-check"></i><b>5</b> Week 4- What is a Genetic Variant?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#main-objectives-3"><i class="fa fa-check"></i><b>5.1</b> Main Objectives</a></li>
<li class="chapter" data-level="5.2" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#to-get-started-lets-dowload-the-data-and-install-a-few-programs"><i class="fa fa-check"></i><b>5.2</b> To get started lets dowload the data and install a few programs</a></li>
<li class="chapter" data-level="5.3" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#raw-read-quality-control"><i class="fa fa-check"></i><b>5.3</b> Raw read quality control</a></li>
<li class="chapter" data-level="5.4" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#trimming-to-remove-adapters"><i class="fa fa-check"></i><b>5.4</b> Trimming to remove adapters</a></li>
<li class="chapter" data-level="5.5" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#building-an-index-of-our-genome"><i class="fa fa-check"></i><b>5.5</b> Building an index of our genome</a></li>
<li class="chapter" data-level="5.6" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#map-reads-to-the-genome"><i class="fa fa-check"></i><b>5.6</b> Map reads to the genome</a></li>
<li class="chapter" data-level="5.7" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#sam-to-bam-file-conversion"><i class="fa fa-check"></i><b>5.7</b> sam to bam file conversion</a></li>
<li class="chapter" data-level="5.8" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#genotype-likelihoods"><i class="fa fa-check"></i><b>5.8</b> Genotype likelihoods</a></li>
<li class="chapter" data-level="5.9" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#group-work-activity--calling-variants-at-a-larger-scale"><i class="fa fa-check"></i><b>5.9</b> Group Work Activity- Calling Variants at a Larger Scale</a></li>
<li class="chapter" data-level="5.10" data-path="week-4--what-is-a-genetic-variant.html"><a href="week-4--what-is-a-genetic-variant.html#key-points-3"><i class="fa fa-check"></i><b>5.10</b> Key Points</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">02-Week2.Rmd</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="week-2--working-with-files" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number"> 3</span> Week 2- Working with Files<a href="week-2--working-with-files.html#week-2--working-with-files" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>We will be following some of the data carpentry tutorial again (Copyright 2016 @ Software Carpentry) “Introduction to the command line for genomics”. We have made some modifications to the data carpentry tutorial to fit our course.</p>
<div id="main-objectives-1" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Main Objectives:<a href="week-2--working-with-files.html#main-objectives-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Learn how to use navigational shortcuts and repeat recently used commands to make coding more efficient</li>
<li>Learn how to view and work with hidden directories and hidden files</li>
<li>Learn how to view and search file contents</li>
</ul>
</div>
<div id="navigational-shortcuts" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Navigational shortcuts<a href="week-2--working-with-files.html#navigational-shortcuts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The root directory is the highest level directory in your file system and contains files that are important for your computer to perform its daily work. While you will be using the root (/) at the beginning of your absolute paths, it is important that you avoid working with data in these higher-level directories, as your commands can permanently alter files that the operating system needs to function. In many cases, trying to run commands in root directories will require special permissions which will be discussed later, so it’s best to avoid them and work within your home directory. Dealing with the home directory is very common. The tilde character, ~, is a shortcut for your home directory. In our case, the root directory is ‘/group’</p>
<p>We can make the ls output more comprehensible by using the flag -F, which tells ls to add a trailing / to the names of directories. First make sure you are in the eve198-genomics directory by printing your working directory</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb1-1"><a href="week-2--working-with-files.html#cb1-1" tabindex="-1"></a>$ cd /group/rbaygrp/eve198-genomics</span>
<span id="cb1-2"><a href="week-2--working-with-files.html#cb1-2" tabindex="-1"></a>$ pwd</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb2-1"><a href="week-2--working-with-files.html#cb2-1" tabindex="-1"></a>/group/rbaygrp/eve198-genomics</span></code></pre></div>
<p>Then type ls -F. It should show your and all your classmates’ directories.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb3-1"><a href="week-2--working-with-files.html#cb3-1" tabindex="-1"></a>$ ls -F</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb4-1"><a href="week-2--working-with-files.html#cb4-1" tabindex="-1"></a>mlarmstrong/</span></code></pre></div>
<p>Anything with a “/” after it is a directory. Things with a “*” after them are programs. If there are no decorations, it’s a file.</p>
<p>Use the <code>-l</code> option for the <code>ls</code> command to display more information for each item in the directory. What is one piece of additional information this long format gives you that you don’t see with the bare <code>ls</code> command?</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb5-1"><a href="week-2--working-with-files.html#cb5-1" tabindex="-1"></a>$ ls -l</span></code></pre></div>
<p>No one can possibly learn all of these arguments, that’s what the manual page is for. You can (and should) refer to the manual page <code>man</code> or other help files as needed.</p>
<p>Let’s navigate to our individual folders:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb6-1"><a href="week-2--working-with-files.html#cb6-1" tabindex="-1"></a>$ cd `yourdirectory`</span>
<span id="cb6-2"><a href="week-2--working-with-files.html#cb6-2" tabindex="-1"></a>$ ls</span></code></pre></div>
<p>Another navigational shortcut that can be really helpful is just hitting the up arrow on your keyboard. This pulls up the line of code that you ran recently! You can keep hitting the up arrow until you get to a line you want. This can help you save time since you won’t need to retype long lines of code over and over again.</p>
<p>Now let’s download some data to work with!</p>
</div>
<div id="details-on-the-fastq-format" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Details on the FASTQ format<a href="week-2--working-with-files.html#details-on-the-fastq-format" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A fastq file is a text file with information on the quality of the sequence data</p>
<p>Although it looks complicated (and it is), it’s easy to understand the
<a href="https://knowledge.illumina.com/software/general/software-general-reference_material-list/000002211">fastq</a> format with a little decoding. Some rules about the format
include…</p>
<table>
<colgroup>
<col width="26%" />
<col width="73%" />
</colgroup>
<thead>
<tr class="header">
<th>Line</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>Always begins with ‘@’ and then information about the read</td>
</tr>
<tr class="even">
<td>2</td>
<td>The actual DNA sequence</td>
</tr>
<tr class="odd">
<td>3</td>
<td>Always begins with a ‘+’ and sometimes the same info in line 1</td>
</tr>
<tr class="even">
<td>4</td>
<td>Has a string of characters which represent the quality (phred) scores; must have same number of characters as line 2</td>
</tr>
</tbody>
</table>
</div>
<div id="our-data-set-fastq-files" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Our data set: FASTQ files<a href="week-2--working-with-files.html#our-data-set-fastq-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can use the command <code>wget</code> which needs a link to the file that we want to download. If there’s a file saved on a website somewhere (anywhere on the internet) <code>wget</code> will download it for you. If our data file is on github, which is where most of our data will be stored, we’ll use the command <code>git-clone</code></p>
<p>In this example we’re going to download all the material in our individual directory. Navigate there and then type the following command:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb7-1"><a href="week-2--working-with-files.html#cb7-1" tabindex="-1"></a>$ wget https://raw.githubusercontent.com/mlarmstrong/IntroGenomics_Data/main/week2.tar.gz</span></code></pre></div>
<p>Use the tar command to uncompress the file. This will also automatically make a week2 directory in your individual directory:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb8-1"><a href="week-2--working-with-files.html#cb8-1" tabindex="-1"></a>tar -xzvf week2.tar.gz</span></code></pre></div>
<p>Check out what is in this directory with the ls command. Remember you can do <code>ls -l</code> to see more information on the contents</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb9-1"><a href="week-2--working-with-files.html#cb9-1" tabindex="-1"></a>untrimmed_fastq sra_metadata TableS2_QTL_Bay_2017.txt</span></code></pre></div>
<p>Now that we know how to navigate around our directory structure, let’s start working with our sequencing files. We did a sequencing experiment and have two results files, which are stored in our <code>untrimmed_fastq</code> directory.</p>
</div>
<div id="wildcards" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Wildcards<a href="week-2--working-with-files.html#wildcards" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Navigate to your <code>untrimmed_fastq</code> directory in your week2 individual directory</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb10-1"><a href="week-2--working-with-files.html#cb10-1" tabindex="-1"></a>$ cd ~/group/rbaygrp/eve198-genomics/yourdirectory/week2/untrimmed_fastq</span></code></pre></div>
<p>We are interested in looking at the FASTQ files in this directory. We can list
all files with the .fastq extension using the command:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb11-1"><a href="week-2--working-with-files.html#cb11-1" tabindex="-1"></a>$ ls *.fastq</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb12-1"><a href="week-2--working-with-files.html#cb12-1" tabindex="-1"></a>SRR097977.fastq  SRR098026.fastq</span></code></pre></div>
<p>The <code>*</code> character is a special type of character called a wildcard, which can be used to represent any number of any type of character.
Thus, <code>*.fastq</code> matches every file that ends with <code>.fastq</code>.</p>
<p>This command:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb13-1"><a href="week-2--working-with-files.html#cb13-1" tabindex="-1"></a>$ ls *977.fastq</span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb14-1"><a href="week-2--working-with-files.html#cb14-1" tabindex="-1"></a>SRR097977.fastq</span></code></pre></div>
<p>lists only the file that ends with <code>977.fastq</code>.</p>
<p>This command:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb15-1"><a href="week-2--working-with-files.html#cb15-1" tabindex="-1"></a>$ ls /usr/bin/*.sh</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb16-1"><a href="week-2--working-with-files.html#cb16-1" tabindex="-1"></a>/usr/bin/gettext.sh  /usr/bin/rescan-scsi-bus.sh</span></code></pre></div>
<p>Lists every file in <code>/usr/bin</code> that ends in the characters <code>.sh</code>.
Note that the output displays <strong>full</strong> paths to files, since
each result starts with <code>/</code>.</p>
<p>This directory contains two files with .fastq extensions. FASTQ is a format for storing information about sequencing reads and their quality. We will be learning more about FASTQ files in a later lesson.</p>
<p><code>echo</code> is a built-in shell command that writes its arguments, like a line of text to standard output.
The <code>echo</code> command can also be used with pattern matching characters, such as wildcard characters.
Here we will use the <code>echo</code> command to see how the wildcard character is interpreted by the shell.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb17-1"><a href="week-2--working-with-files.html#cb17-1" tabindex="-1"></a>$ echo *.fastq</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb18-1"><a href="week-2--working-with-files.html#cb18-1" tabindex="-1"></a>SRR097977.fastq SRR098026.fastq</span></code></pre></div>
<p>The <code>*</code> is expanded to include any file that ends with <code>.fastq</code>. We can see that the output of
<code>echo *.fastq</code> is the same as that of <code>ls *.fastq</code>.</p>
<p>What would the output look like if the wildcard could <em>not</em> be matched? Compare the outputs of
<code>echo *.missing</code> and <code>ls *.missing</code>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb19-1"><a href="week-2--working-with-files.html#cb19-1" tabindex="-1"></a>$ echo *.missing</span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb20-1"><a href="week-2--working-with-files.html#cb20-1" tabindex="-1"></a>*.missing</span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb21-1"><a href="week-2--working-with-files.html#cb21-1" tabindex="-1"></a>$ ls *.missing</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb22-1"><a href="week-2--working-with-files.html#cb22-1" tabindex="-1"></a>ls: cannot access &#39;*.missing&#39;: No such file or directory</span></code></pre></div>
<blockquote>
<h2 id="class-exercise-1" class="hasAnchor">Class Exercise 1<a href="week-2--working-with-files.html#class-exercise-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Do each of the following tasks from your current directory using a single
<code>ls</code> command for each:</p>
<ol style="list-style-type: decimal">
<li>List all of the files in <code>/usr/bin</code> that start with the letter ‘c’.</li>
<li>List all of the files in <code>/usr/bin</code> that contain the letter ‘a’.</li>
<li>List all of the files in <code>/usr/bin</code> that end with the letter ‘o’.</li>
</ol>
<p>Bonus: List all of the files in <code>/usr/bin</code> that contain the letter ‘a’ or the
letter ‘c’.</p>
<p>Hint: The bonus question requires a Unix wildcard that we haven’t talked about
yet. Try searching the internet for information about Unix wildcards to find
what you need to solve the bonus problem.</p>
</blockquote>
</div>
<div id="command-history" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Command History<a href="week-2--working-with-files.html#command-history" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If you want to repeat a command that you’ve run recently, you can access previous
commands using the up arrow on your keyboard to go back to the most recent
command. Likewise, the down arrow takes you forward in the command history.</p>
<p>A few more useful shortcuts:</p>
<ul>
<li><kbd>Ctrl</kbd>+<kbd>C</kbd> will cancel the command you are writing, and give you a
fresh prompt.</li>
<li><kbd>Ctrl</kbd>+<kbd>R</kbd> will do a reverse-search through your command history. This
is very useful.</li>
<li><kbd>Ctrl</kbd>+<kbd>L</kbd> or the <code>clear</code> command will clear your screen.</li>
</ul>
<p>You can also review your recent commands with the <code>history</code> command, by entering:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb23-1"><a href="week-2--working-with-files.html#cb23-1" tabindex="-1"></a>$ history</span></code></pre></div>
<p>to see a numbered list of recent commands. You can reuse one of these commands
directly by referring to the number of that command.</p>
<p>For example, if your history looked like this:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb24-1"><a href="week-2--working-with-files.html#cb24-1" tabindex="-1"></a>259  ls *</span>
<span id="cb24-2"><a href="week-2--working-with-files.html#cb24-2" tabindex="-1"></a>260  ls /usr/bin/*.sh</span>
<span id="cb24-3"><a href="week-2--working-with-files.html#cb24-3" tabindex="-1"></a>261  ls *R1*fastq</span></code></pre></div>
<p>then you could repeat command #260 by entering:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb25-1"><a href="week-2--working-with-files.html#cb25-1" tabindex="-1"></a>$ !260</span></code></pre></div>
<p>Type <code>!</code> (exclamation point) and then the number of the command from your history.
You will be glad you learned this when you need to re-run very complicated commands.
For more information on advanced usage of <code>history</code>, read section 9.3 of
<a href="https://www.gnu.org/software/bash/manual/html_node/index.html">Bash manual</a>.</p>
</div>
<div id="examining-files" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> Examining Files<a href="week-2--working-with-files.html#examining-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We now know how to switch directories, run programs, and look at the
contents of directories, but how do we look at the contents of files?</p>
<p>One way to examine a file is to print out all of the
contents using the program <code>cat</code>.</p>
<p>Enter the following command from within the <code>untrimmed_fastq</code> directory:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb26-1"><a href="week-2--working-with-files.html#cb26-1" tabindex="-1"></a>$ cat SRR098026.fastq</span></code></pre></div>
<p>This will print out all of the contents of the <code>SRR098026.fastq</code> to the screen.</p>
<blockquote>
<h2 id="class-exercise-2" class="hasAnchor">Class Exercise 2<a href="week-2--working-with-files.html#class-exercise-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>Print out the contents of the <code>~/group/rbaygrp/eve198-genomics/'yourdirectory'/Week2/untrimmed_fastq/SRR097977.fastq</code> file. What is the last line of the file?</li>
<li>From your home directory, and without changing directories,
use one short command to print the contents of all of the files in
the <code>group/rbaygrp/eve198-genomics/yourdirectory/week2/untrimmed_fastq</code> directory.</li>
</ol>
</blockquote>
<p><code>cat</code> is a terrific program, but when the file is really big, it can
be annoying to use. The program, <code>less</code>, is useful for this
case. <code>less</code> opens the file as read only, and lets you navigate through it. The navigation commands
are identical to the <code>man</code> program.</p>
<p>Enter the following command:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb27-1"><a href="week-2--working-with-files.html#cb27-1" tabindex="-1"></a>$ less SRR097977.fastq</span></code></pre></div>
<p>Some navigation commands in <code>less</code>:</p>
<table>
<thead>
<tr class="header">
<th>key</th>
<th>action</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><kbd>Space</kbd></td>
<td>to go forward</td>
</tr>
<tr class="even">
<td><kbd>b</kbd></td>
<td>to go backward</td>
</tr>
<tr class="odd">
<td><kbd>g</kbd></td>
<td>to go to the beginning</td>
</tr>
<tr class="even">
<td><kbd>G</kbd></td>
<td>to go to the end</td>
</tr>
<tr class="odd">
<td><kbd>q</kbd></td>
<td>to quit</td>
</tr>
</tbody>
</table>
<p><code>less</code> also gives you a way of searching through files. Use the
“/” key to begin a search. Enter the word you would like
to search for and press <code>enter</code>. The screen will jump to the next location where
that word is found.</p>
<p><strong>Shortcut:</strong> If you hit “/” then “enter”, <code>less</code> will repeat
the previous search. <code>less</code> searches from the current location and
works its way forward. Scroll up a couple lines on your terminal to verify
you are at the beginning of the file. Note, if you are at the end of the file and search
for the sequence “CAA”, <code>less</code> will not find it. You either need to go to the
beginning of the file (by typing <code>g</code>) and search again using <code>/</code> or you
can use <code>?</code> to search backwards in the same way you used <code>/</code> previously.</p>
<p>For instance, let’s search forward for the sequence <code>TTTTT</code> in our file.
You can see that we go right to that sequence, what it looks like,
and where it is in the file. If you continue to type <code>/</code> and hit return, you will move
forward to the next instance of this sequence motif. If you instead type <code>?</code> and hit
return, you will search backwards and move up the file to previous examples of this motif.</p>
<blockquote>
<h2 id="class-exercise-3" class="hasAnchor">Class Exercise 3<a href="week-2--working-with-files.html#class-exercise-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>What are the next three nucleotides (characters) after the first instance of the sequence quoted above?</p>
</blockquote>
<p>Remember, the <code>man</code> program actually uses <code>less</code> internally and
therefore uses the same commands, so you can search documentation
using “/” as well!</p>
<p>There’s another way that we can look at files, and in this case, just
look at part of them. This can be particularly useful if we just want
to see the beginning or end of the file, or see how it’s formatted.</p>
<p>The commands are <code>head</code> and <code>tail</code> and they let you look at
the beginning and end of a file, respectively.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb28-1"><a href="week-2--working-with-files.html#cb28-1" tabindex="-1"></a>$ head SRR098026.fastq</span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb29-1"><a href="week-2--working-with-files.html#cb29-1" tabindex="-1"></a>@SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb29-2"><a href="week-2--working-with-files.html#cb29-2" tabindex="-1"></a>NNNNNNNNNNNNNNNNCNNNNNNNNNNNNNNNNNN</span>
<span id="cb29-3"><a href="week-2--working-with-files.html#cb29-3" tabindex="-1"></a>+SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb29-4"><a href="week-2--working-with-files.html#cb29-4" tabindex="-1"></a>!!!!!!!!!!!!!!!!#!!!!!!!!!!!!!!!!!!</span>
<span id="cb29-5"><a href="week-2--working-with-files.html#cb29-5" tabindex="-1"></a>@SRR098026.2 HWUSI-EAS1599_1:2:1:0:312 length=35</span>
<span id="cb29-6"><a href="week-2--working-with-files.html#cb29-6" tabindex="-1"></a>NNNNNNNNNNNNNNNNANNNNNNNNNNNNNNNNNN</span>
<span id="cb29-7"><a href="week-2--working-with-files.html#cb29-7" tabindex="-1"></a>+SRR098026.2 HWUSI-EAS1599_1:2:1:0:312 length=35</span>
<span id="cb29-8"><a href="week-2--working-with-files.html#cb29-8" tabindex="-1"></a>!!!!!!!!!!!!!!!!#!!!!!!!!!!!!!!!!!!</span>
<span id="cb29-9"><a href="week-2--working-with-files.html#cb29-9" tabindex="-1"></a>@SRR098026.3 HWUSI-EAS1599_1:2:1:0:570 length=35</span>
<span id="cb29-10"><a href="week-2--working-with-files.html#cb29-10" tabindex="-1"></a>NNNNNNNNNNNNNNNNANNNNNNNNNNNNNNNNNN</span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb30-1"><a href="week-2--working-with-files.html#cb30-1" tabindex="-1"></a>$ tail SRR098026.fastq</span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb31-1"><a href="week-2--working-with-files.html#cb31-1" tabindex="-1"></a>+SRR098026.247 HWUSI-EAS1599_1:2:1:2:1311 length=35</span>
<span id="cb31-2"><a href="week-2--working-with-files.html#cb31-2" tabindex="-1"></a>#!##!#################!!!!!!!######</span>
<span id="cb31-3"><a href="week-2--working-with-files.html#cb31-3" tabindex="-1"></a>@SRR098026.248 HWUSI-EAS1599_1:2:1:2:118 length=35</span>
<span id="cb31-4"><a href="week-2--working-with-files.html#cb31-4" tabindex="-1"></a>GNTGNGGTCATCATACGCGCCCNNNNNNNGGCATG</span>
<span id="cb31-5"><a href="week-2--working-with-files.html#cb31-5" tabindex="-1"></a>+SRR098026.248 HWUSI-EAS1599_1:2:1:2:118 length=35</span>
<span id="cb31-6"><a href="week-2--working-with-files.html#cb31-6" tabindex="-1"></a>B!;?!A=5922:##########!!!!!!!######</span>
<span id="cb31-7"><a href="week-2--working-with-files.html#cb31-7" tabindex="-1"></a>@SRR098026.249 HWUSI-EAS1599_1:2:1:2:1057 length=35</span>
<span id="cb31-8"><a href="week-2--working-with-files.html#cb31-8" tabindex="-1"></a>CNCTNTATGCGTACGGCAGTGANNNNNNNGGAGAT</span>
<span id="cb31-9"><a href="week-2--working-with-files.html#cb31-9" tabindex="-1"></a>+SRR098026.249 HWUSI-EAS1599_1:2:1:2:1057 length=35</span>
<span id="cb31-10"><a href="week-2--working-with-files.html#cb31-10" tabindex="-1"></a>A!@B!BBB@ABAB#########!!!!!!!######</span></code></pre></div>
<p>The <code>-n</code> option to either of these commands can be used to print the
first or last <code>n</code> lines of a file.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb32-1"><a href="week-2--working-with-files.html#cb32-1" tabindex="-1"></a>$ head -n 1 SRR098026.fastq</span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb33-1"><a href="week-2--working-with-files.html#cb33-1" tabindex="-1"></a>@SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb34-1"><a href="week-2--working-with-files.html#cb34-1" tabindex="-1"></a>$ tail -n 1 SRR098026.fastq</span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb35-1"><a href="week-2--working-with-files.html#cb35-1" tabindex="-1"></a>A!@B!BBB@ABAB#########!!!!!!!######</span></code></pre></div>
<p>We can view the first complete read in one of the files in our dataset by using <code>head</code> to look at
the first four lines.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb36-1"><a href="week-2--working-with-files.html#cb36-1" tabindex="-1"></a>$ head -n 4 SRR098026.fastq</span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb37-1"><a href="week-2--working-with-files.html#cb37-1" tabindex="-1"></a>@SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb37-2"><a href="week-2--working-with-files.html#cb37-2" tabindex="-1"></a>NNNNNNNNNNNNNNNNCNNNNNNNNNNNNNNNNNN</span>
<span id="cb37-3"><a href="week-2--working-with-files.html#cb37-3" tabindex="-1"></a>+SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb37-4"><a href="week-2--working-with-files.html#cb37-4" tabindex="-1"></a>!!!!!!!!!!!!!!!!#!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<p>All but one of the nucleotides in this read are unknown (<code>N</code>). This is a pretty bad read!</p>
<p>Line 4 shows the quality for each nucleotide in the read. Quality is interpreted as the
probability of an incorrect base call (e.g. 1 in 10) or, equivalently, the base call
accuracy (e.g. 90%). To make it possible to line up each individual nucleotide with its quality
score, the numerical score is converted into a code where each individual character
represents the numerical quality score for an individual nucleotide. For example, in the line
above, the quality score line is:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb38-1"><a href="week-2--working-with-files.html#cb38-1" tabindex="-1"></a>!!!!!!!!!!!!!!!!#!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<p>The <code>#</code> character and each of the <code>!</code> characters represent the encoded quality for an
individual nucleotide. The numerical value assigned to each of these characters depends on the
sequencing platform that generated the reads. The sequencing machine used to generate our data
uses the standard Sanger quality PHRED score encoding, Illumina version 1.8 onwards.</p>
<p>Here is a link showing what those different symbols mean for quality scores: <a href="https://help.basespace.illumina.com/files-used-by-basespace/quality-scores" class="uri">https://help.basespace.illumina.com/files-used-by-basespace/quality-scores</a></p>
<p>Each character is assigned a quality score between 0 and 42 as shown in the chart below.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb39-1"><a href="week-2--working-with-files.html#cb39-1" tabindex="-1"></a>Quality encoding: !&quot;#$%<span class="er">&amp;</span>&#39;()*+,-./0123456789:;<span class="er">&lt;</span>=&gt;?@ABCDEFGHIJK</span>
<span id="cb39-2"><a href="week-2--working-with-files.html#cb39-2" tabindex="-1"></a>                  |         |         |         |         |</span>
<span id="cb39-3"><a href="week-2--working-with-files.html#cb39-3" tabindex="-1"></a>Quality score:    0........10........20........30........40..                          </span></code></pre></div>
<p>Each quality score represents the probability that the corresponding nucleotide call is
incorrect. This quality score is logarithmically based, so a quality score of 10 reflects a
base call accuracy of 90%, but a quality score of 20 reflects a base call accuracy of 99%.
These probability values are the results from the base calling algorithm and dependent on how
much signal was captured for the base incorporation.</p>
<p>Looking back at our read:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb40-1"><a href="week-2--working-with-files.html#cb40-1" tabindex="-1"></a>@SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb40-2"><a href="week-2--working-with-files.html#cb40-2" tabindex="-1"></a>NNNNNNNNNNNNNNNNCNNNNNNNNNNNNNNNNNN</span>
<span id="cb40-3"><a href="week-2--working-with-files.html#cb40-3" tabindex="-1"></a>+SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb40-4"><a href="week-2--working-with-files.html#cb40-4" tabindex="-1"></a>!!!!!!!!!!!!!!!!#!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<p>we can now see that the quality of each of the <code>N</code>s is 0 and the quality of the only
nucleotide call (<code>C</code>) is also very poor (<code>#</code> = a quality score of 2). This is indeed a very
bad read.</p>
</div>
<div id="group-work-activity--examining-a-fastq-file" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> Group Work Activity- Examining a Fastq File<a href="week-2--working-with-files.html#group-work-activity--examining-a-fastq-file" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Navigate to eve198-genomics and copy the ‘CCGPMC004_M0D060025C_S150_L003_R1_001.fastq.gz’ file in the week2_activity directory to your individual directory.</p>
<p>Next you will need to unzip the fastq file. To do this we will use the command “gunzip”. This unzips a gzipped file. “gzip” zips a file again. It will take a second since this is a larger file than our previous examples. It is actually an urchin sample from a recent genomics project Rachael and I worked on!</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb41-1"><a href="week-2--working-with-files.html#cb41-1" tabindex="-1"></a>gunzip CCGPMC004_M0D060025C_S150_L003_R1_001.fastq.gz</span></code></pre></div>
<p>Then answer the following questions and submit them to the “Week 2: Examining Files” assignment on canvas. I would not recommend using ‘less’ or ‘cat’ on this file due to its size.</p>
<ol style="list-style-type: decimal">
<li>Write out the first two lines of CCGPMC004_M0D060025C_S150_L003_R1_001.fastq</li>
<li>Remember that line 4 in a fastq file shows the quality (phred) scores. If you look at line 4 and reference the score table from the link and figures below. What is the most common symbol, Q-score and error probability for most of our nucleotide calls?</li>
</ol>
<p><a href="https://help.basespace.illumina.com/files-used-by-basespace/quality-scores" class="uri">https://help.basespace.illumina.com/files-used-by-basespace/quality-scores</a>
<img src="figs/QC-errortable.png" alt="QC Error Table" />
<img src="figs/scoretable1.jpg" alt="Score Table" />
<img src="figs/scoretable2.jpg" alt="Score Table" /></p>
</div>
<div id="key-points-1" class="section level2 hasAnchor" number="3.9">
<h2><span class="header-section-number">3.9</span> Key Points<a href="week-2--working-with-files.html#key-points-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p>The /, ~, and .. characters represent important navigational shortcuts.</p></li>
<li><p>The <code>history</code> command and the up arrow on your keyboard can be used to repeat recently used commands.</p></li>
<li><p>You can view file contents using <code>less</code>, <code>cat</code>, <code>head</code> or <code>tail</code>.</p></li>
</ul>
<details>
<summary>
<span style="color: orange;">Class Exercise Solutions</span>
</summary>
<p>
<blockquote>
<blockquote>
<h2 id="exercise-1-solution" class="hasAnchor">Exercise 1: Solution<a href="week-2--working-with-files.html#exercise-1-solution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><code>ls /usr/bin/c*</code></li>
<li><code>ls /usr/bin/*a*</code></li>
<li><code>ls /usr/bin/*o</code><br />
Bonus: <code>ls /usr/bin/*[ac]*</code></li>
</ol>
<h2 id="exercise-2-solution" class="hasAnchor">Exercise 2: Solution<a href="week-2--working-with-files.html#exercise-2-solution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>The last line of the file is <code>C:CCC::CCCCCCCC&lt;8?6A:C28C&lt;608'&amp;&amp;&amp;,'$</code>.</li>
<li><code>cat ./~/group/rbaygrp/eve198-genomics/yourdirectory/week2/untrimmed_fastq/*</code>
## Exercise 3: Solution
<code>CAC</code></li>
</ol>
</blockquote>
</blockquote>
</p>
</details>
<p> </p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="week-1--welcome.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="week-3--working-with-files-part-2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
